// react-infinite-canvas/dist/react-infinite-canvas.es.js
(function(){"use strict";try{if(typeof document<"u"){var e=document.createElement("style");e.appendChild(document.createTextNode("._dotSvgContainer_64xbd_1{position:absolute;width:100%;height:100%;inset-block-start:0;inset-inline-start:0}._container_1xxyl_1{position:relative;overflow:hidden;width:100%;height:100%}._canvasWrapper_1xxyl_8{position:relative;width:100%;height:100%;overflow:hidden;background-color:var(--kf-color-bg-gray);cursor:grab;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}._canvasWrapper_1xxyl_8._panning_1xxyl_21{cursor:grabbing}._canvas_1xxyl_8{position:relative;z-index:2;width:100%;height:100%}._canvas_1xxyl_8>div:first-of-type{position:absolute;inset-inline-start:0;inset-block-start:0;transition:transform .01s ease}._contentWrapper_1xxyl_39{position:absolute}._verticalScrollBar_17thb_1{position:absolute;inset-inline-end:2px;inset-block-start:0;inset-block-end:0;background:transparent}._horizontalScrollBar_17thb_9{position:absolute;inset-block-end:2px;inset-inline-start:0;inset-inline-end:0;background:transparent}._verticalScrollBar_17thb_1>div,._horizontalScrollBar_17thb_9>div{position:absolute;border-radius:50px;z-index:8;cursor:pointer}._verticalScrollBar_17thb_1>div{width:100%}._horizontalScrollBar_17thb_9>div{height:100%}")),document.head.appendChild(e)}}catch(t){console.error("vite-plugin-css-injected-by-js",t)}})();
import _e, { forwardRef as Nn, useRef as ft, useState as zn, useImperativeHandle as Pn, useEffect as Mn, useCallback as jt, Fragment as Er, memo as Rr, useMemo as In } from "react";
var ze = { exports: {} }, Mt = {};
/*
object-assign
(c) Sindre Sorhus
@license MIT
*/
var Te, an;
function Dn() {
  if (an)
    return Te;
  an = 1;
  var t = Object.getOwnPropertySymbols, e = Object.prototype.hasOwnProperty, n = Object.prototype.propertyIsEnumerable;
  function r(a) {
    if (a == null)
      throw new TypeError("Object.assign cannot be called with null or undefined");
    return Object(a);
  }
  function i() {
    try {
      if (!Object.assign)
        return !1;
      var a = new String("abc");
      if (a[5] = "de", Object.getOwnPropertyNames(a)[0] === "5")
        return !1;
      for (var o = {}, s = 0; s < 10; s++)
        o["_" + String.fromCharCode(s)] = s;
      var l = Object.getOwnPropertyNames(o).map(function(c) {
        return o[c];
      });
      if (l.join("") !== "0123456789")
        return !1;
      var h = {};
      return "abcdefghijklmnopqrst".split("").forEach(function(c) {
        h[c] = c;
      }), Object.keys(Object.assign({}, h)).join("") === "abcdefghijklmnopqrst";
    } catch {
      return !1;
    }
  }
  return Te = i() ? Object.assign : function(a, o) {
    for (var s, l = r(a), h, c = 1; c < arguments.length; c++) {
      s = Object(arguments[c]);
      for (var _ in s)
        e.call(s, _) && (l[_] = s[_]);
      if (t) {
        h = t(s);
        for (var p = 0; p < h.length; p++)
          n.call(s, h[p]) && (l[h[p]] = s[h[p]]);
      }
    }
    return l;
  }, Te;
}
/** @license React v17.0.2
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var on;
function Tr() {
  if (on)
    return Mt;
  on = 1, Dn();
  var t = _e, e = 60103;
  if (Mt.Fragment = 60107, typeof Symbol == "function" && Symbol.for) {
    var n = Symbol.for;
    e = n("react.element"), Mt.Fragment = n("react.fragment");
  }
  var r = t.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, i = Object.prototype.hasOwnProperty, a = { key: !0, ref: !0, __self: !0, __source: !0 };
  function o(s, l, h) {
    var c, _ = {}, p = null, y = null;
    h !== void 0 && (p = "" + h), l.key !== void 0 && (p = "" + l.key), l.ref !== void 0 && (y = l.ref);
    for (c in l)
      i.call(l, c) && !a.hasOwnProperty(c) && (_[c] = l[c]);
    if (s && s.defaultProps)
      for (c in l = s.defaultProps, l)
        _[c] === void 0 && (_[c] = l[c]);
    return { $$typeof: e, type: s, key: p, ref: y, props: _, _owner: r.current };
  }
  return Mt.jsx = o, Mt.jsxs = o, Mt;
}
var Se = {};
/** @license React v17.0.2
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var sn;
function Sr() {
  return sn || (sn = 1, function(t) {
    process.env.NODE_ENV !== "production" && function() {
      var e = _e, n = Dn(), r = 60103, i = 60106;
      t.Fragment = 60107;
      var a = 60108, o = 60114, s = 60109, l = 60110, h = 60112, c = 60113, _ = 60120, p = 60115, y = 60116, O = 60121, T = 60122, M = 60117, w = 60129, $ = 60131;
      if (typeof Symbol == "function" && Symbol.for) {
        var C = Symbol.for;
        r = C("react.element"), i = C("react.portal"), t.Fragment = C("react.fragment"), a = C("react.strict_mode"), o = C("react.profiler"), s = C("react.provider"), l = C("react.context"), h = C("react.forward_ref"), c = C("react.suspense"), _ = C("react.suspense_list"), p = C("react.memo"), y = C("react.lazy"), O = C("react.block"), T = C("react.server.block"), M = C("react.fundamental"), C("react.scope"), C("react.opaque.id"), w = C("react.debug_trace_mode"), C("react.offscreen"), $ = C("react.legacy_hidden");
      }
      var V = typeof Symbol == "function" && Symbol.iterator, J = "@@iterator";
      function j(u) {
        if (u === null || typeof u != "object")
          return null;
        var g = V && u[V] || u[J];
        return typeof g == "function" ? g : null;
      }
      var I = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
      function A(u) {
        {
          for (var g = arguments.length, x = new Array(g > 1 ? g - 1 : 0), S = 1; S < g; S++)
            x[S - 1] = arguments[S];
          H("error", u, x);
        }
      }
      function H(u, g, x) {
        {
          var S = I.ReactDebugCurrentFrame, Y = S.getStackAddendum();
          Y !== "" && (g += "%s", x = x.concat([Y]));
          var F = x.map(function(k) {
            return "" + k;
          });
          F.unshift("Warning: " + g), Function.prototype.apply.call(console[u], console, F);
        }
      }
      var rt = !1;
      function ut(u) {
        return !!(typeof u == "string" || typeof u == "function" || u === t.Fragment || u === o || u === w || u === a || u === c || u === _ || u === $ || rt || typeof u == "object" && u !== null && (u.$$typeof === y || u.$$typeof === p || u.$$typeof === s || u.$$typeof === l || u.$$typeof === h || u.$$typeof === M || u.$$typeof === O || u[0] === T));
      }
      function P(u, g, x) {
        var S = g.displayName || g.name || "";
        return u.displayName || (S !== "" ? x + "(" + S + ")" : x);
      }
      function L(u) {
        return u.displayName || "Context";
      }
      function f(u) {
        if (u == null)
          return null;
        if (typeof u.tag == "number" && A("Received an unexpected object in getComponentName(). This is likely a bug in React. Please file an issue."), typeof u == "function")
          return u.displayName || u.name || null;
        if (typeof u == "string")
          return u;
        switch (u) {
          case t.Fragment:
            return "Fragment";
          case i:
            return "Portal";
          case o:
            return "Profiler";
          case a:
            return "StrictMode";
          case c:
            return "Suspense";
          case _:
            return "SuspenseList";
        }
        if (typeof u == "object")
          switch (u.$$typeof) {
            case l:
              var g = u;
              return L(g) + ".Consumer";
            case s:
              var x = u;
              return L(x._context) + ".Provider";
            case h:
              return P(u, u.render, "ForwardRef");
            case p:
              return f(u.type);
            case O:
              return f(u._render);
            case y: {
              var S = u, Y = S._payload, F = S._init;
              try {
                return f(F(Y));
              } catch {
                return null;
              }
            }
          }
        return null;
      }
      var v = 0, d, m, b, E, R, z, N;
      function q() {
      }
      q.__reactDisabledLog = !0;
      function X() {
        {
          if (v === 0) {
            d = console.log, m = console.info, b = console.warn, E = console.error, R = console.group, z = console.groupCollapsed, N = console.groupEnd;
            var u = {
              configurable: !0,
              enumerable: !0,
              value: q,
              writable: !0
            };
            Object.defineProperties(console, {
              info: u,
              log: u,
              warn: u,
              error: u,
              group: u,
              groupCollapsed: u,
              groupEnd: u
            });
          }
          v++;
        }
      }
      function W() {
        {
          if (v--, v === 0) {
            var u = {
              configurable: !0,
              enumerable: !0,
              writable: !0
            };
            Object.defineProperties(console, {
              log: n({}, u, {
                value: d
              }),
              info: n({}, u, {
                value: m
              }),
              warn: n({}, u, {
                value: b
              }),
              error: n({}, u, {
                value: E
              }),
              group: n({}, u, {
                value: R
              }),
              groupCollapsed: n({}, u, {
                value: z
              }),
              groupEnd: n({}, u, {
                value: N
              })
            });
          }
          v < 0 && A("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
        }
      }
      var K = I.ReactCurrentDispatcher, B;
      function Q(u, g, x) {
        {
          if (B === void 0)
            try {
              throw Error();
            } catch (Y) {
              var S = Y.stack.trim().match(/\n( *(at )?)/);
              B = S && S[1] || "";
            }
          return `
` + B + u;
        }
      }
      var it = !1, yt;
      {
        var Nt = typeof WeakMap == "function" ? WeakMap : Map;
        yt = new Nt();
      }
      function bt(u, g) {
        if (!u || it)
          return "";
        {
          var x = yt.get(u);
          if (x !== void 0)
            return x;
        }
        var S;
        it = !0;
        var Y = Error.prepareStackTrace;
        Error.prepareStackTrace = void 0;
        var F;
        F = K.current, K.current = null, X();
        try {
          if (g) {
            var k = function() {
              throw Error();
            };
            if (Object.defineProperty(k.prototype, "props", {
              set: function() {
                throw Error();
              }
            }), typeof Reflect == "object" && Reflect.construct) {
              try {
                Reflect.construct(k, []);
              } catch (_t) {
                S = _t;
              }
              Reflect.construct(u, [], k);
            } else {
              try {
                k.call();
              } catch (_t) {
                S = _t;
              }
              u.call(k.prototype);
            }
          } else {
            try {
              throw Error();
            } catch (_t) {
              S = _t;
            }
            u();
          }
        } catch (_t) {
          if (_t && S && typeof _t.stack == "string") {
            for (var D = _t.stack.split(`
`), et = S.stack.split(`
`), G = D.length - 1, Z = et.length - 1; G >= 1 && Z >= 0 && D[G] !== et[Z]; )
              Z--;
            for (; G >= 1 && Z >= 0; G--, Z--)
              if (D[G] !== et[Z]) {
                if (G !== 1 || Z !== 1)
                  do
                    if (G--, Z--, Z < 0 || D[G] !== et[Z]) {
                      var pt = `
` + D[G].replace(" at new ", " at ");
                      return typeof u == "function" && yt.set(u, pt), pt;
                    }
                  while (G >= 1 && Z >= 0);
                break;
              }
          }
        } finally {
          it = !1, K.current = F, W(), Error.prepareStackTrace = Y;
        }
        var Pt = u ? u.displayName || u.name : "", rn = Pt ? Q(Pt) : "";
        return typeof u == "function" && yt.set(u, rn), rn;
      }
      function Et(u, g, x) {
        return bt(u, !1);
      }
      function ye(u) {
        var g = u.prototype;
        return !!(g && g.isReactComponent);
      }
      function Rt(u, g, x) {
        if (u == null)
          return "";
        if (typeof u == "function")
          return bt(u, ye(u));
        if (typeof u == "string")
          return Q(u);
        switch (u) {
          case c:
            return Q("Suspense");
          case _:
            return Q("SuspenseList");
        }
        if (typeof u == "object")
          switch (u.$$typeof) {
            case h:
              return Et(u.render);
            case p:
              return Rt(u.type, g, x);
            case O:
              return Et(u._render);
            case y: {
              var S = u, Y = S._payload, F = S._init;
              try {
                return Rt(F(Y), g, x);
              } catch {
              }
            }
          }
        return "";
      }
      var $t = {}, Jt = I.ReactDebugCurrentFrame;
      function Tt(u) {
        if (u) {
          var g = u._owner, x = Rt(u.type, u._source, g ? g.type : null);
          Jt.setExtraStackFrame(x);
        } else
          Jt.setExtraStackFrame(null);
      }
      function Qt(u, g, x, S, Y) {
        {
          var F = Function.call.bind(Object.prototype.hasOwnProperty);
          for (var k in u)
            if (F(u, k)) {
              var D = void 0;
              try {
                if (typeof u[k] != "function") {
                  var et = Error((S || "React class") + ": " + x + " type `" + k + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof u[k] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                  throw et.name = "Invariant Violation", et;
                }
                D = u[k](g, k, S, x, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
              } catch (G) {
                D = G;
              }
              D && !(D instanceof Error) && (Tt(Y), A("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", S || "React class", x, k, typeof D), Tt(null)), D instanceof Error && !(D.message in $t) && ($t[D.message] = !0, Tt(Y), A("Failed %s type: %s", x, D.message), Tt(null));
            }
        }
      }
      var Lt = I.ReactCurrentOwner, xe = Object.prototype.hasOwnProperty, sr = {
        key: !0,
        ref: !0,
        __self: !0,
        __source: !0
      }, Ke, Be, we;
      we = {};
      function ur(u) {
        if (xe.call(u, "ref")) {
          var g = Object.getOwnPropertyDescriptor(u, "ref").get;
          if (g && g.isReactWarning)
            return !1;
        }
        return u.ref !== void 0;
      }
      function cr(u) {
        if (xe.call(u, "key")) {
          var g = Object.getOwnPropertyDescriptor(u, "key").get;
          if (g && g.isReactWarning)
            return !1;
        }
        return u.key !== void 0;
      }
      function lr(u, g) {
        if (typeof u.ref == "string" && Lt.current && g && Lt.current.stateNode !== g) {
          var x = f(Lt.current.type);
          we[x] || (A('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', f(Lt.current.type), u.ref), we[x] = !0);
        }
      }
      function fr(u, g) {
        {
          var x = function() {
            Ke || (Ke = !0, A("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", g));
          };
          x.isReactWarning = !0, Object.defineProperty(u, "key", {
            get: x,
            configurable: !0
          });
        }
      }
      function hr(u, g) {
        {
          var x = function() {
            Be || (Be = !0, A("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", g));
          };
          x.isReactWarning = !0, Object.defineProperty(u, "ref", {
            get: x,
            configurable: !0
          });
        }
      }
      var dr = function(u, g, x, S, Y, F, k) {
        var D = {
          // This tag allows us to uniquely identify this as a React Element
          $$typeof: r,
          // Built-in properties that belong on the element
          type: u,
          key: g,
          ref: x,
          props: k,
          // Record the component responsible for creating this element.
          _owner: F
        };
        return D._store = {}, Object.defineProperty(D._store, "validated", {
          configurable: !1,
          enumerable: !1,
          writable: !0,
          value: !1
        }), Object.defineProperty(D, "_self", {
          configurable: !1,
          enumerable: !1,
          writable: !1,
          value: S
        }), Object.defineProperty(D, "_source", {
          configurable: !1,
          enumerable: !1,
          writable: !1,
          value: Y
        }), Object.freeze && (Object.freeze(D.props), Object.freeze(D)), D;
      };
      function pr(u, g, x, S, Y) {
        {
          var F, k = {}, D = null, et = null;
          x !== void 0 && (D = "" + x), cr(g) && (D = "" + g.key), ur(g) && (et = g.ref, lr(g, Y));
          for (F in g)
            xe.call(g, F) && !sr.hasOwnProperty(F) && (k[F] = g[F]);
          if (u && u.defaultProps) {
            var G = u.defaultProps;
            for (F in G)
              k[F] === void 0 && (k[F] = G[F]);
          }
          if (D || et) {
            var Z = typeof u == "function" ? u.displayName || u.name || "Unknown" : u;
            D && fr(k, Z), et && hr(k, Z);
          }
          return dr(u, D, et, Y, S, Lt.current, k);
        }
      }
      var be = I.ReactCurrentOwner, Ze = I.ReactDebugCurrentFrame;
      function zt(u) {
        if (u) {
          var g = u._owner, x = Rt(u.type, u._source, g ? g.type : null);
          Ze.setExtraStackFrame(x);
        } else
          Ze.setExtraStackFrame(null);
      }
      var Ee;
      Ee = !1;
      function Re(u) {
        return typeof u == "object" && u !== null && u.$$typeof === r;
      }
      function Je() {
        {
          if (be.current) {
            var u = f(be.current.type);
            if (u)
              return `

Check the render method of \`` + u + "`.";
          }
          return "";
        }
      }
      function _r(u) {
        {
          if (u !== void 0) {
            var g = u.fileName.replace(/^.*[\\\/]/, ""), x = u.lineNumber;
            return `

Check your code at ` + g + ":" + x + ".";
          }
          return "";
        }
      }
      var Qe = {};
      function gr(u) {
        {
          var g = Je();
          if (!g) {
            var x = typeof u == "string" ? u : u.displayName || u.name;
            x && (g = `

Check the top-level render call using <` + x + ">.");
          }
          return g;
        }
      }
      function tn(u, g) {
        {
          if (!u._store || u._store.validated || u.key != null)
            return;
          u._store.validated = !0;
          var x = gr(g);
          if (Qe[x])
            return;
          Qe[x] = !0;
          var S = "";
          u && u._owner && u._owner !== be.current && (S = " It was passed a child from " + f(u._owner.type) + "."), zt(u), A('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', x, S), zt(null);
        }
      }
      function en(u, g) {
        {
          if (typeof u != "object")
            return;
          if (Array.isArray(u))
            for (var x = 0; x < u.length; x++) {
              var S = u[x];
              Re(S) && tn(S, g);
            }
          else if (Re(u))
            u._store && (u._store.validated = !0);
          else if (u) {
            var Y = j(u);
            if (typeof Y == "function" && Y !== u.entries)
              for (var F = Y.call(u), k; !(k = F.next()).done; )
                Re(k.value) && tn(k.value, g);
          }
        }
      }
      function vr(u) {
        {
          var g = u.type;
          if (g == null || typeof g == "string")
            return;
          var x;
          if (typeof g == "function")
            x = g.propTypes;
          else if (typeof g == "object" && (g.$$typeof === h || // Note: Memo only checks outer props here.
          // Inner props are checked in the reconciler.
          g.$$typeof === p))
            x = g.propTypes;
          else
            return;
          if (x) {
            var S = f(g);
            Qt(x, u.props, "prop", S, u);
          } else if (g.PropTypes !== void 0 && !Ee) {
            Ee = !0;
            var Y = f(g);
            A("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", Y || "Unknown");
          }
          typeof g.getDefaultProps == "function" && !g.getDefaultProps.isReactClassApproved && A("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
        }
      }
      function mr(u) {
        {
          for (var g = Object.keys(u.props), x = 0; x < g.length; x++) {
            var S = g[x];
            if (S !== "children" && S !== "key") {
              zt(u), A("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", S), zt(null);
              break;
            }
          }
          u.ref !== null && (zt(u), A("Invalid attribute `ref` supplied to `React.Fragment`."), zt(null));
        }
      }
      function nn(u, g, x, S, Y, F) {
        {
          var k = ut(u);
          if (!k) {
            var D = "";
            (u === void 0 || typeof u == "object" && u !== null && Object.keys(u).length === 0) && (D += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
            var et = _r(Y);
            et ? D += et : D += Je();
            var G;
            u === null ? G = "null" : Array.isArray(u) ? G = "array" : u !== void 0 && u.$$typeof === r ? (G = "<" + (f(u.type) || "Unknown") + " />", D = " Did you accidentally export a JSX literal instead of a component?") : G = typeof u, A("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", G, D);
          }
          var Z = pr(u, g, x, Y, F);
          if (Z == null)
            return Z;
          if (k) {
            var pt = g.children;
            if (pt !== void 0)
              if (S)
                if (Array.isArray(pt)) {
                  for (var Pt = 0; Pt < pt.length; Pt++)
                    en(pt[Pt], u);
                  Object.freeze && Object.freeze(pt);
                } else
                  A("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
              else
                en(pt, u);
          }
          return u === t.Fragment ? mr(Z) : vr(Z), Z;
        }
      }
      function yr(u, g, x) {
        return nn(u, g, x, !0);
      }
      function xr(u, g, x) {
        return nn(u, g, x, !1);
      }
      var wr = xr, br = yr;
      t.jsx = wr, t.jsxs = br;
    }();
  }(Se)), Se;
}
process.env.NODE_ENV === "production" ? ze.exports = Tr() : ze.exports = Sr();
var U = ze.exports, Pe = "http://www.w3.org/1999/xhtml";
const un = {
  svg: "http://www.w3.org/2000/svg",
  xhtml: Pe,
  xlink: "http://www.w3.org/1999/xlink",
  xml: "http://www.w3.org/XML/1998/namespace",
  xmlns: "http://www.w3.org/2000/xmlns/"
};
function ge(t) {
  var e = t += "", n = e.indexOf(":");
  return n >= 0 && (e = t.slice(0, n)) !== "xmlns" && (t = t.slice(n + 1)), un.hasOwnProperty(e) ? { space: un[e], local: t } : t;
}
function Cr(t) {
  return function() {
    var e = this.ownerDocument, n = this.namespaceURI;
    return n === Pe && e.documentElement.namespaceURI === Pe ? e.createElement(t) : e.createElementNS(n, t);
  };
}
function Or(t) {
  return function() {
    return this.ownerDocument.createElementNS(t.space, t.local);
  };
}
function kn(t) {
  var e = ge(t);
  return (e.local ? Or : Cr)(e);
}
function Ar() {
}
function He(t) {
  return t == null ? Ar : function() {
    return this.querySelector(t);
  };
}
function Nr(t) {
  typeof t != "function" && (t = He(t));
  for (var e = this._groups, n = e.length, r = new Array(n), i = 0; i < n; ++i)
    for (var a = e[i], o = a.length, s = r[i] = new Array(o), l, h, c = 0; c < o; ++c)
      (l = a[c]) && (h = t.call(l, l.__data__, c, a)) && ("__data__" in l && (h.__data__ = l.__data__), s[c] = h);
  return new ot(r, this._parents);
}
function zr(t) {
  return t == null ? [] : Array.isArray(t) ? t : Array.from(t);
}
function Pr() {
  return [];
}
function $n(t) {
  return t == null ? Pr : function() {
    return this.querySelectorAll(t);
  };
}
function Mr(t) {
  return function() {
    return zr(t.apply(this, arguments));
  };
}
function Ir(t) {
  typeof t == "function" ? t = Mr(t) : t = $n(t);
  for (var e = this._groups, n = e.length, r = [], i = [], a = 0; a < n; ++a)
    for (var o = e[a], s = o.length, l, h = 0; h < s; ++h)
      (l = o[h]) && (r.push(t.call(l, l.__data__, h, o)), i.push(l));
  return new ot(r, i);
}
function Ln(t) {
  return function() {
    return this.matches(t);
  };
}
function Yn(t) {
  return function(e) {
    return e.matches(t);
  };
}
var Dr = Array.prototype.find;
function kr(t) {
  return function() {
    return Dr.call(this.children, t);
  };
}
function $r() {
  return this.firstElementChild;
}
function Lr(t) {
  return this.select(t == null ? $r : kr(typeof t == "function" ? t : Yn(t)));
}
var Yr = Array.prototype.filter;
function Fr() {
  return Array.from(this.children);
}
function Xr(t) {
  return function() {
    return Yr.call(this.children, t);
  };
}
function Hr(t) {
  return this.selectAll(t == null ? Fr : Xr(typeof t == "function" ? t : Yn(t)));
}
function jr(t) {
  typeof t != "function" && (t = Ln(t));
  for (var e = this._groups, n = e.length, r = new Array(n), i = 0; i < n; ++i)
    for (var a = e[i], o = a.length, s = r[i] = [], l, h = 0; h < o; ++h)
      (l = a[h]) && t.call(l, l.__data__, h, a) && s.push(l);
  return new ot(r, this._parents);
}
function Fn(t) {
  return new Array(t.length);
}
function Ur() {
  return new ot(this._enter || this._groups.map(Fn), this._parents);
}
function ue(t, e) {
  this.ownerDocument = t.ownerDocument, this.namespaceURI = t.namespaceURI, this._next = null, this._parent = t, this.__data__ = e;
}
ue.prototype = {
  constructor: ue,
  appendChild: function(t) {
    return this._parent.insertBefore(t, this._next);
  },
  insertBefore: function(t, e) {
    return this._parent.insertBefore(t, e);
  },
  querySelector: function(t) {
    return this._parent.querySelector(t);
  },
  querySelectorAll: function(t) {
    return this._parent.querySelectorAll(t);
  }
};
function Vr(t) {
  return function() {
    return t;
  };
}
function qr(t, e, n, r, i, a) {
  for (var o = 0, s, l = e.length, h = a.length; o < h; ++o)
    (s = e[o]) ? (s.__data__ = a[o], r[o] = s) : n[o] = new ue(t, a[o]);
  for (; o < l; ++o)
    (s = e[o]) && (i[o] = s);
}
function Wr(t, e, n, r, i, a, o) {
  var s, l, h = /* @__PURE__ */ new Map(), c = e.length, _ = a.length, p = new Array(c), y;
  for (s = 0; s < c; ++s)
    (l = e[s]) && (p[s] = y = o.call(l, l.__data__, s, e) + "", h.has(y) ? i[s] = l : h.set(y, l));
  for (s = 0; s < _; ++s)
    y = o.call(t, a[s], s, a) + "", (l = h.get(y)) ? (r[s] = l, l.__data__ = a[s], h.delete(y)) : n[s] = new ue(t, a[s]);
  for (s = 0; s < c; ++s)
    (l = e[s]) && h.get(p[s]) === l && (i[s] = l);
}
function Gr(t) {
  return t.__data__;
}
function Kr(t, e) {
  if (!arguments.length)
    return Array.from(this, Gr);
  var n = e ? Wr : qr, r = this._parents, i = this._groups;
  typeof t != "function" && (t = Vr(t));
  for (var a = i.length, o = new Array(a), s = new Array(a), l = new Array(a), h = 0; h < a; ++h) {
    var c = r[h], _ = i[h], p = _.length, y = Br(t.call(c, c && c.__data__, h, r)), O = y.length, T = s[h] = new Array(O), M = o[h] = new Array(O), w = l[h] = new Array(p);
    n(c, _, T, M, w, y, e);
    for (var $ = 0, C = 0, V, J; $ < O; ++$)
      if (V = T[$]) {
        for ($ >= C && (C = $ + 1); !(J = M[C]) && ++C < O; )
          ;
        V._next = J || null;
      }
  }
  return o = new ot(o, r), o._enter = s, o._exit = l, o;
}
function Br(t) {
  return typeof t == "object" && "length" in t ? t : Array.from(t);
}
function Zr() {
  return new ot(this._exit || this._groups.map(Fn), this._parents);
}
function Jr(t, e, n) {
  var r = this.enter(), i = this, a = this.exit();
  return typeof t == "function" ? (r = t(r), r && (r = r.selection())) : r = r.append(t + ""), e != null && (i = e(i), i && (i = i.selection())), n == null ? a.remove() : n(a), r && i ? r.merge(i).order() : i;
}
function Qr(t) {
  for (var e = t.selection ? t.selection() : t, n = this._groups, r = e._groups, i = n.length, a = r.length, o = Math.min(i, a), s = new Array(i), l = 0; l < o; ++l)
    for (var h = n[l], c = r[l], _ = h.length, p = s[l] = new Array(_), y, O = 0; O < _; ++O)
      (y = h[O] || c[O]) && (p[O] = y);
  for (; l < i; ++l)
    s[l] = n[l];
  return new ot(s, this._parents);
}
function ti() {
  for (var t = this._groups, e = -1, n = t.length; ++e < n; )
    for (var r = t[e], i = r.length - 1, a = r[i], o; --i >= 0; )
      (o = r[i]) && (a && o.compareDocumentPosition(a) ^ 4 && a.parentNode.insertBefore(o, a), a = o);
  return this;
}
function ei(t) {
  t || (t = ni);
  function e(_, p) {
    return _ && p ? t(_.__data__, p.__data__) : !_ - !p;
  }
  for (var n = this._groups, r = n.length, i = new Array(r), a = 0; a < r; ++a) {
    for (var o = n[a], s = o.length, l = i[a] = new Array(s), h, c = 0; c < s; ++c)
      (h = o[c]) && (l[c] = h);
    l.sort(e);
  }
  return new ot(i, this._parents).order();
}
function ni(t, e) {
  return t < e ? -1 : t > e ? 1 : t >= e ? 0 : NaN;
}
function ri() {
  var t = arguments[0];
  return arguments[0] = this, t.apply(null, arguments), this;
}
function ii() {
  return Array.from(this);
}
function ai() {
  for (var t = this._groups, e = 0, n = t.length; e < n; ++e)
    for (var r = t[e], i = 0, a = r.length; i < a; ++i) {
      var o = r[i];
      if (o)
        return o;
    }
  return null;
}
function oi() {
  let t = 0;
  for (const e of this)
    ++t;
  return t;
}
function si() {
  return !this.node();
}
function ui(t) {
  for (var e = this._groups, n = 0, r = e.length; n < r; ++n)
    for (var i = e[n], a = 0, o = i.length, s; a < o; ++a)
      (s = i[a]) && t.call(s, s.__data__, a, i);
  return this;
}
function ci(t) {
  return function() {
    this.removeAttribute(t);
  };
}
function li(t) {
  return function() {
    this.removeAttributeNS(t.space, t.local);
  };
}
function fi(t, e) {
  return function() {
    this.setAttribute(t, e);
  };
}
function hi(t, e) {
  return function() {
    this.setAttributeNS(t.space, t.local, e);
  };
}
function di(t, e) {
  return function() {
    var n = e.apply(this, arguments);
    n == null ? this.removeAttribute(t) : this.setAttribute(t, n);
  };
}
function pi(t, e) {
  return function() {
    var n = e.apply(this, arguments);
    n == null ? this.removeAttributeNS(t.space, t.local) : this.setAttributeNS(t.space, t.local, n);
  };
}
function _i(t, e) {
  var n = ge(t);
  if (arguments.length < 2) {
    var r = this.node();
    return n.local ? r.getAttributeNS(n.space, n.local) : r.getAttribute(n);
  }
  return this.each((e == null ? n.local ? li : ci : typeof e == "function" ? n.local ? pi : di : n.local ? hi : fi)(n, e));
}
function Xn(t) {
  return t.ownerDocument && t.ownerDocument.defaultView || t.document && t || t.defaultView;
}
function gi(t) {
  return function() {
    this.style.removeProperty(t);
  };
}
function vi(t, e, n) {
  return function() {
    this.style.setProperty(t, e, n);
  };
}
function mi(t, e, n) {
  return function() {
    var r = e.apply(this, arguments);
    r == null ? this.style.removeProperty(t) : this.style.setProperty(t, r, n);
  };
}
function yi(t, e, n) {
  return arguments.length > 1 ? this.each((e == null ? gi : typeof e == "function" ? mi : vi)(t, e, n ?? "")) : Dt(this.node(), t);
}
function Dt(t, e) {
  return t.style.getPropertyValue(e) || Xn(t).getComputedStyle(t, null).getPropertyValue(e);
}
function xi(t) {
  return function() {
    delete this[t];
  };
}
function wi(t, e) {
  return function() {
    this[t] = e;
  };
}
function bi(t, e) {
  return function() {
    var n = e.apply(this, arguments);
    n == null ? delete this[t] : this[t] = n;
  };
}
function Ei(t, e) {
  return arguments.length > 1 ? this.each((e == null ? xi : typeof e == "function" ? bi : wi)(t, e)) : this.node()[t];
}
function Hn(t) {
  return t.trim().split(/^|\s+/);
}
function je(t) {
  return t.classList || new jn(t);
}
function jn(t) {
  this._node = t, this._names = Hn(t.getAttribute("class") || "");
}
jn.prototype = {
  add: function(t) {
    var e = this._names.indexOf(t);
    e < 0 && (this._names.push(t), this._node.setAttribute("class", this._names.join(" ")));
  },
  remove: function(t) {
    var e = this._names.indexOf(t);
    e >= 0 && (this._names.splice(e, 1), this._node.setAttribute("class", this._names.join(" ")));
  },
  contains: function(t) {
    return this._names.indexOf(t) >= 0;
  }
};
function Un(t, e) {
  for (var n = je(t), r = -1, i = e.length; ++r < i; )
    n.add(e[r]);
}
function Vn(t, e) {
  for (var n = je(t), r = -1, i = e.length; ++r < i; )
    n.remove(e[r]);
}
function Ri(t) {
  return function() {
    Un(this, t);
  };
}
function Ti(t) {
  return function() {
    Vn(this, t);
  };
}
function Si(t, e) {
  return function() {
    (e.apply(this, arguments) ? Un : Vn)(this, t);
  };
}
function Ci(t, e) {
  var n = Hn(t + "");
  if (arguments.length < 2) {
    for (var r = je(this.node()), i = -1, a = n.length; ++i < a; )
      if (!r.contains(n[i]))
        return !1;
    return !0;
  }
  return this.each((typeof e == "function" ? Si : e ? Ri : Ti)(n, e));
}
function Oi() {
  this.textContent = "";
}
function Ai(t) {
  return function() {
    this.textContent = t;
  };
}
function Ni(t) {
  return function() {
    var e = t.apply(this, arguments);
    this.textContent = e ?? "";
  };
}
function zi(t) {
  return arguments.length ? this.each(t == null ? Oi : (typeof t == "function" ? Ni : Ai)(t)) : this.node().textContent;
}
function Pi() {
  this.innerHTML = "";
}
function Mi(t) {
  return function() {
    this.innerHTML = t;
  };
}
function Ii(t) {
  return function() {
    var e = t.apply(this, arguments);
    this.innerHTML = e ?? "";
  };
}
function Di(t) {
  return arguments.length ? this.each(t == null ? Pi : (typeof t == "function" ? Ii : Mi)(t)) : this.node().innerHTML;
}
function ki() {
  this.nextSibling && this.parentNode.appendChild(this);
}
function $i() {
  return this.each(ki);
}
function Li() {
  this.previousSibling && this.parentNode.insertBefore(this, this.parentNode.firstChild);
}
function Yi() {
  return this.each(Li);
}
function Fi(t) {
  var e = typeof t == "function" ? t : kn(t);
  return this.select(function() {
    return this.appendChild(e.apply(this, arguments));
  });
}
function Xi() {
  return null;
}
function Hi(t, e) {
  var n = typeof t == "function" ? t : kn(t), r = e == null ? Xi : typeof e == "function" ? e : He(e);
  return this.select(function() {
    return this.insertBefore(n.apply(this, arguments), r.apply(this, arguments) || null);
  });
}
function ji() {
  var t = this.parentNode;
  t && t.removeChild(this);
}
function Ui() {
  return this.each(ji);
}
function Vi() {
  var t = this.cloneNode(!1), e = this.parentNode;
  return e ? e.insertBefore(t, this.nextSibling) : t;
}
function qi() {
  var t = this.cloneNode(!0), e = this.parentNode;
  return e ? e.insertBefore(t, this.nextSibling) : t;
}
function Wi(t) {
  return this.select(t ? qi : Vi);
}
function Gi(t) {
  return arguments.length ? this.property("__data__", t) : this.node().__data__;
}
function Ki(t) {
  return function(e) {
    t.call(this, e, this.__data__);
  };
}
function Bi(t) {
  return t.trim().split(/^|\s+/).map(function(e) {
    var n = "", r = e.indexOf(".");
    return r >= 0 && (n = e.slice(r + 1), e = e.slice(0, r)), { type: e, name: n };
  });
}
function Zi(t) {
  return function() {
    var e = this.__on;
    if (e) {
      for (var n = 0, r = -1, i = e.length, a; n < i; ++n)
        a = e[n], (!t.type || a.type === t.type) && a.name === t.name ? this.removeEventListener(a.type, a.listener, a.options) : e[++r] = a;
      ++r ? e.length = r : delete this.__on;
    }
  };
}
function Ji(t, e, n) {
  return function() {
    var r = this.__on, i, a = Ki(e);
    if (r) {
      for (var o = 0, s = r.length; o < s; ++o)
        if ((i = r[o]).type === t.type && i.name === t.name) {
          this.removeEventListener(i.type, i.listener, i.options), this.addEventListener(i.type, i.listener = a, i.options = n), i.value = e;
          return;
        }
    }
    this.addEventListener(t.type, a, n), i = { type: t.type, name: t.name, value: e, listener: a, options: n }, r ? r.push(i) : this.__on = [i];
  };
}
function Qi(t, e, n) {
  var r = Bi(t + ""), i, a = r.length, o;
  if (arguments.length < 2) {
    var s = this.node().__on;
    if (s) {
      for (var l = 0, h = s.length, c; l < h; ++l)
        for (i = 0, c = s[l]; i < a; ++i)
          if ((o = r[i]).type === c.type && o.name === c.name)
            return c.value;
    }
    return;
  }
  for (s = e ? Ji : Zi, i = 0; i < a; ++i)
    this.each(s(r[i], e, n));
  return this;
}
function qn(t, e, n) {
  var r = Xn(t), i = r.CustomEvent;
  typeof i == "function" ? i = new i(e, n) : (i = r.document.createEvent("Event"), n ? (i.initEvent(e, n.bubbles, n.cancelable), i.detail = n.detail) : i.initEvent(e, !1, !1)), t.dispatchEvent(i);
}
function ta(t, e) {
  return function() {
    return qn(this, t, e);
  };
}
function ea(t, e) {
  return function() {
    return qn(this, t, e.apply(this, arguments));
  };
}
function na(t, e) {
  return this.each((typeof e == "function" ? ea : ta)(t, e));
}
function* ra() {
  for (var t = this._groups, e = 0, n = t.length; e < n; ++e)
    for (var r = t[e], i = 0, a = r.length, o; i < a; ++i)
      (o = r[i]) && (yield o);
}
var Wn = [null];
function ot(t, e) {
  this._groups = t, this._parents = e;
}
function Bt() {
  return new ot([[document.documentElement]], Wn);
}
function ia() {
  return this;
}
ot.prototype = Bt.prototype = {
  constructor: ot,
  select: Nr,
  selectAll: Ir,
  selectChild: Lr,
  selectChildren: Hr,
  filter: jr,
  data: Kr,
  enter: Ur,
  exit: Zr,
  join: Jr,
  merge: Qr,
  selection: ia,
  order: ti,
  sort: ei,
  call: ri,
  nodes: ii,
  node: ai,
  size: oi,
  empty: si,
  each: ui,
  attr: _i,
  style: yi,
  property: Ei,
  classed: Ci,
  text: zi,
  html: Di,
  raise: $i,
  lower: Yi,
  append: Fi,
  insert: Hi,
  remove: Ui,
  clone: Wi,
  datum: Gi,
  on: Qi,
  dispatch: na,
  [Symbol.iterator]: ra
};
function nt(t) {
  return typeof t == "string" ? new ot([[document.querySelector(t)]], [document.documentElement]) : new ot([[t]], Wn);
}
function aa(t) {
  let e;
  for (; e = t.sourceEvent; )
    t = e;
  return t;
}
function xt(t, e) {
  if (t = aa(t), e === void 0 && (e = t.currentTarget), e) {
    var n = e.ownerSVGElement || e;
    if (n.createSVGPoint) {
      var r = n.createSVGPoint();
      return r.x = t.clientX, r.y = t.clientY, r = r.matrixTransform(e.getScreenCTM().inverse()), [r.x, r.y];
    }
    if (e.getBoundingClientRect) {
      var i = e.getBoundingClientRect();
      return [t.clientX - i.left - e.clientLeft, t.clientY - i.top - e.clientTop];
    }
  }
  return [t.pageX, t.pageY];
}
var oa = { value: () => {
} };
function Ue() {
  for (var t = 0, e = arguments.length, n = {}, r; t < e; ++t) {
    if (!(r = arguments[t] + "") || r in n || /[\s.]/.test(r))
      throw new Error("illegal type: " + r);
    n[r] = [];
  }
  return new ie(n);
}
function ie(t) {
  this._ = t;
}
function sa(t, e) {
  return t.trim().split(/^|\s+/).map(function(n) {
    var r = "", i = n.indexOf(".");
    if (i >= 0 && (r = n.slice(i + 1), n = n.slice(0, i)), n && !e.hasOwnProperty(n))
      throw new Error("unknown type: " + n);
    return { type: n, name: r };
  });
}
ie.prototype = Ue.prototype = {
  constructor: ie,
  on: function(t, e) {
    var n = this._, r = sa(t + "", n), i, a = -1, o = r.length;
    if (arguments.length < 2) {
      for (; ++a < o; )
        if ((i = (t = r[a]).type) && (i = ua(n[i], t.name)))
          return i;
      return;
    }
    if (e != null && typeof e != "function")
      throw new Error("invalid callback: " + e);
    for (; ++a < o; )
      if (i = (t = r[a]).type)
        n[i] = cn(n[i], t.name, e);
      else if (e == null)
        for (i in n)
          n[i] = cn(n[i], t.name, null);
    return this;
  },
  copy: function() {
    var t = {}, e = this._;
    for (var n in e)
      t[n] = e[n].slice();
    return new ie(t);
  },
  call: function(t, e) {
    if ((i = arguments.length - 2) > 0)
      for (var n = new Array(i), r = 0, i, a; r < i; ++r)
        n[r] = arguments[r + 2];
    if (!this._.hasOwnProperty(t))
      throw new Error("unknown type: " + t);
    for (a = this._[t], r = 0, i = a.length; r < i; ++r)
      a[r].value.apply(e, n);
  },
  apply: function(t, e, n) {
    if (!this._.hasOwnProperty(t))
      throw new Error("unknown type: " + t);
    for (var r = this._[t], i = 0, a = r.length; i < a; ++i)
      r[i].value.apply(e, n);
  }
};
function ua(t, e) {
  for (var n = 0, r = t.length, i; n < r; ++n)
    if ((i = t[n]).name === e)
      return i.value;
}
function cn(t, e, n) {
  for (var r = 0, i = t.length; r < i; ++r)
    if (t[r].name === e) {
      t[r] = oa, t = t.slice(0, r).concat(t.slice(r + 1));
      break;
    }
  return n != null && t.push({ name: e, value: n }), t;
}
const Me = { capture: !0, passive: !1 };
function Ie(t) {
  t.preventDefault(), t.stopImmediatePropagation();
}
function ca(t) {
  var e = t.document.documentElement, n = nt(t).on("dragstart.drag", Ie, Me);
  "onselectstart" in e ? n.on("selectstart.drag", Ie, Me) : (e.__noselect = e.style.MozUserSelect, e.style.MozUserSelect = "none");
}
function la(t, e) {
  var n = t.document.documentElement, r = nt(t).on("dragstart.drag", null);
  e && (r.on("click.drag", Ie, Me), setTimeout(function() {
    r.on("click.drag", null);
  }, 0)), "onselectstart" in n ? r.on("selectstart.drag", null) : (n.style.MozUserSelect = n.__noselect, delete n.__noselect);
}
function Ve(t, e, n) {
  t.prototype = e.prototype = n, n.constructor = t;
}
function Gn(t, e) {
  var n = Object.create(t.prototype);
  for (var r in e)
    n[r] = e[r];
  return n;
}
function Zt() {
}
var qt = 0.7, ce = 1 / qt, It = "\\s*([+-]?\\d+)\\s*", Wt = "\\s*([+-]?(?:\\d*\\.)?\\d+(?:[eE][+-]?\\d+)?)\\s*", ht = "\\s*([+-]?(?:\\d*\\.)?\\d+(?:[eE][+-]?\\d+)?)%\\s*", fa = /^#([0-9a-f]{3,8})$/, ha = new RegExp(`^rgb\\(${It},${It},${It}\\)$`), da = new RegExp(`^rgb\\(${ht},${ht},${ht}\\)$`), pa = new RegExp(`^rgba\\(${It},${It},${It},${Wt}\\)$`), _a = new RegExp(`^rgba\\(${ht},${ht},${ht},${Wt}\\)$`), ga = new RegExp(`^hsl\\(${Wt},${ht},${ht}\\)$`), va = new RegExp(`^hsla\\(${Wt},${ht},${ht},${Wt}\\)$`), ln = {
  aliceblue: 15792383,
  antiquewhite: 16444375,
  aqua: 65535,
  aquamarine: 8388564,
  azure: 15794175,
  beige: 16119260,
  bisque: 16770244,
  black: 0,
  blanchedalmond: 16772045,
  blue: 255,
  blueviolet: 9055202,
  brown: 10824234,
  burlywood: 14596231,
  cadetblue: 6266528,
  chartreuse: 8388352,
  chocolate: 13789470,
  coral: 16744272,
  cornflowerblue: 6591981,
  cornsilk: 16775388,
  crimson: 14423100,
  cyan: 65535,
  darkblue: 139,
  darkcyan: 35723,
  darkgoldenrod: 12092939,
  darkgray: 11119017,
  darkgreen: 25600,
  darkgrey: 11119017,
  darkkhaki: 12433259,
  darkmagenta: 9109643,
  darkolivegreen: 5597999,
  darkorange: 16747520,
  darkorchid: 10040012,
  darkred: 9109504,
  darksalmon: 15308410,
  darkseagreen: 9419919,
  darkslateblue: 4734347,
  darkslategray: 3100495,
  darkslategrey: 3100495,
  darkturquoise: 52945,
  darkviolet: 9699539,
  deeppink: 16716947,
  deepskyblue: 49151,
  dimgray: 6908265,
  dimgrey: 6908265,
  dodgerblue: 2003199,
  firebrick: 11674146,
  floralwhite: 16775920,
  forestgreen: 2263842,
  fuchsia: 16711935,
  gainsboro: 14474460,
  ghostwhite: 16316671,
  gold: 16766720,
  goldenrod: 14329120,
  gray: 8421504,
  green: 32768,
  greenyellow: 11403055,
  grey: 8421504,
  honeydew: 15794160,
  hotpink: 16738740,
  indianred: 13458524,
  indigo: 4915330,
  ivory: 16777200,
  khaki: 15787660,
  lavender: 15132410,
  lavenderblush: 16773365,
  lawngreen: 8190976,
  lemonchiffon: 16775885,
  lightblue: 11393254,
  lightcoral: 15761536,
  lightcyan: 14745599,
  lightgoldenrodyellow: 16448210,
  lightgray: 13882323,
  lightgreen: 9498256,
  lightgrey: 13882323,
  lightpink: 16758465,
  lightsalmon: 16752762,
  lightseagreen: 2142890,
  lightskyblue: 8900346,
  lightslategray: 7833753,
  lightslategrey: 7833753,
  lightsteelblue: 11584734,
  lightyellow: 16777184,
  lime: 65280,
  limegreen: 3329330,
  linen: 16445670,
  magenta: 16711935,
  maroon: 8388608,
  mediumaquamarine: 6737322,
  mediumblue: 205,
  mediumorchid: 12211667,
  mediumpurple: 9662683,
  mediumseagreen: 3978097,
  mediumslateblue: 8087790,
  mediumspringgreen: 64154,
  mediumturquoise: 4772300,
  mediumvioletred: 13047173,
  midnightblue: 1644912,
  mintcream: 16121850,
  mistyrose: 16770273,
  moccasin: 16770229,
  navajowhite: 16768685,
  navy: 128,
  oldlace: 16643558,
  olive: 8421376,
  olivedrab: 7048739,
  orange: 16753920,
  orangered: 16729344,
  orchid: 14315734,
  palegoldenrod: 15657130,
  palegreen: 10025880,
  paleturquoise: 11529966,
  palevioletred: 14381203,
  papayawhip: 16773077,
  peachpuff: 16767673,
  peru: 13468991,
  pink: 16761035,
  plum: 14524637,
  powderblue: 11591910,
  purple: 8388736,
  rebeccapurple: 6697881,
  red: 16711680,
  rosybrown: 12357519,
  royalblue: 4286945,
  saddlebrown: 9127187,
  salmon: 16416882,
  sandybrown: 16032864,
  seagreen: 3050327,
  seashell: 16774638,
  sienna: 10506797,
  silver: 12632256,
  skyblue: 8900331,
  slateblue: 6970061,
  slategray: 7372944,
  slategrey: 7372944,
  snow: 16775930,
  springgreen: 65407,
  steelblue: 4620980,
  tan: 13808780,
  teal: 32896,
  thistle: 14204888,
  tomato: 16737095,
  turquoise: 4251856,
  violet: 15631086,
  wheat: 16113331,
  white: 16777215,
  whitesmoke: 16119285,
  yellow: 16776960,
  yellowgreen: 10145074
};
Ve(Zt, Gt, {
  copy(t) {
    return Object.assign(new this.constructor(), this, t);
  },
  displayable() {
    return this.rgb().displayable();
  },
  hex: fn,
  // Deprecated! Use color.formatHex.
  formatHex: fn,
  formatHex8: ma,
  formatHsl: ya,
  formatRgb: hn,
  toString: hn
});
function fn() {
  return this.rgb().formatHex();
}
function ma() {
  return this.rgb().formatHex8();
}
function ya() {
  return Kn(this).formatHsl();
}
function hn() {
  return this.rgb().formatRgb();
}
function Gt(t) {
  var e, n;
  return t = (t + "").trim().toLowerCase(), (e = fa.exec(t)) ? (n = e[1].length, e = parseInt(e[1], 16), n === 6 ? dn(e) : n === 3 ? new at(e >> 8 & 15 | e >> 4 & 240, e >> 4 & 15 | e & 240, (e & 15) << 4 | e & 15, 1) : n === 8 ? te(e >> 24 & 255, e >> 16 & 255, e >> 8 & 255, (e & 255) / 255) : n === 4 ? te(e >> 12 & 15 | e >> 8 & 240, e >> 8 & 15 | e >> 4 & 240, e >> 4 & 15 | e & 240, ((e & 15) << 4 | e & 15) / 255) : null) : (e = ha.exec(t)) ? new at(e[1], e[2], e[3], 1) : (e = da.exec(t)) ? new at(e[1] * 255 / 100, e[2] * 255 / 100, e[3] * 255 / 100, 1) : (e = pa.exec(t)) ? te(e[1], e[2], e[3], e[4]) : (e = _a.exec(t)) ? te(e[1] * 255 / 100, e[2] * 255 / 100, e[3] * 255 / 100, e[4]) : (e = ga.exec(t)) ? gn(e[1], e[2] / 100, e[3] / 100, 1) : (e = va.exec(t)) ? gn(e[1], e[2] / 100, e[3] / 100, e[4]) : ln.hasOwnProperty(t) ? dn(ln[t]) : t === "transparent" ? new at(NaN, NaN, NaN, 0) : null;
}
function dn(t) {
  return new at(t >> 16 & 255, t >> 8 & 255, t & 255, 1);
}
function te(t, e, n, r) {
  return r <= 0 && (t = e = n = NaN), new at(t, e, n, r);
}
function xa(t) {
  return t instanceof Zt || (t = Gt(t)), t ? (t = t.rgb(), new at(t.r, t.g, t.b, t.opacity)) : new at();
}
function De(t, e, n, r) {
  return arguments.length === 1 ? xa(t) : new at(t, e, n, r ?? 1);
}
function at(t, e, n, r) {
  this.r = +t, this.g = +e, this.b = +n, this.opacity = +r;
}
Ve(at, De, Gn(Zt, {
  brighter(t) {
    return t = t == null ? ce : Math.pow(ce, t), new at(this.r * t, this.g * t, this.b * t, this.opacity);
  },
  darker(t) {
    return t = t == null ? qt : Math.pow(qt, t), new at(this.r * t, this.g * t, this.b * t, this.opacity);
  },
  rgb() {
    return this;
  },
  clamp() {
    return new at(Ot(this.r), Ot(this.g), Ot(this.b), le(this.opacity));
  },
  displayable() {
    return -0.5 <= this.r && this.r < 255.5 && -0.5 <= this.g && this.g < 255.5 && -0.5 <= this.b && this.b < 255.5 && 0 <= this.opacity && this.opacity <= 1;
  },
  hex: pn,
  // Deprecated! Use color.formatHex.
  formatHex: pn,
  formatHex8: wa,
  formatRgb: _n,
  toString: _n
}));
function pn() {
  return `#${Ct(this.r)}${Ct(this.g)}${Ct(this.b)}`;
}
function wa() {
  return `#${Ct(this.r)}${Ct(this.g)}${Ct(this.b)}${Ct((isNaN(this.opacity) ? 1 : this.opacity) * 255)}`;
}
function _n() {
  const t = le(this.opacity);
  return `${t === 1 ? "rgb(" : "rgba("}${Ot(this.r)}, ${Ot(this.g)}, ${Ot(this.b)}${t === 1 ? ")" : `, ${t})`}`;
}
function le(t) {
  return isNaN(t) ? 1 : Math.max(0, Math.min(1, t));
}
function Ot(t) {
  return Math.max(0, Math.min(255, Math.round(t) || 0));
}
function Ct(t) {
  return t = Ot(t), (t < 16 ? "0" : "") + t.toString(16);
}
function gn(t, e, n, r) {
  return r <= 0 ? t = e = n = NaN : n <= 0 || n >= 1 ? t = e = NaN : e <= 0 && (t = NaN), new ct(t, e, n, r);
}
function Kn(t) {
  if (t instanceof ct)
    return new ct(t.h, t.s, t.l, t.opacity);
  if (t instanceof Zt || (t = Gt(t)), !t)
    return new ct();
  if (t instanceof ct)
    return t;
  t = t.rgb();
  var e = t.r / 255, n = t.g / 255, r = t.b / 255, i = Math.min(e, n, r), a = Math.max(e, n, r), o = NaN, s = a - i, l = (a + i) / 2;
  return s ? (e === a ? o = (n - r) / s + (n < r) * 6 : n === a ? o = (r - e) / s + 2 : o = (e - n) / s + 4, s /= l < 0.5 ? a + i : 2 - a - i, o *= 60) : s = l > 0 && l < 1 ? 0 : o, new ct(o, s, l, t.opacity);
}
function ba(t, e, n, r) {
  return arguments.length === 1 ? Kn(t) : new ct(t, e, n, r ?? 1);
}
function ct(t, e, n, r) {
  this.h = +t, this.s = +e, this.l = +n, this.opacity = +r;
}
Ve(ct, ba, Gn(Zt, {
  brighter(t) {
    return t = t == null ? ce : Math.pow(ce, t), new ct(this.h, this.s, this.l * t, this.opacity);
  },
  darker(t) {
    return t = t == null ? qt : Math.pow(qt, t), new ct(this.h, this.s, this.l * t, this.opacity);
  },
  rgb() {
    var t = this.h % 360 + (this.h < 0) * 360, e = isNaN(t) || isNaN(this.s) ? 0 : this.s, n = this.l, r = n + (n < 0.5 ? n : 1 - n) * e, i = 2 * n - r;
    return new at(
      Ce(t >= 240 ? t - 240 : t + 120, i, r),
      Ce(t, i, r),
      Ce(t < 120 ? t + 240 : t - 120, i, r),
      this.opacity
    );
  },
  clamp() {
    return new ct(vn(this.h), ee(this.s), ee(this.l), le(this.opacity));
  },
  displayable() {
    return (0 <= this.s && this.s <= 1 || isNaN(this.s)) && 0 <= this.l && this.l <= 1 && 0 <= this.opacity && this.opacity <= 1;
  },
  formatHsl() {
    const t = le(this.opacity);
    return `${t === 1 ? "hsl(" : "hsla("}${vn(this.h)}, ${ee(this.s) * 100}%, ${ee(this.l) * 100}%${t === 1 ? ")" : `, ${t})`}`;
  }
}));
function vn(t) {
  return t = (t || 0) % 360, t < 0 ? t + 360 : t;
}
function ee(t) {
  return Math.max(0, Math.min(1, t || 0));
}
function Ce(t, e, n) {
  return (t < 60 ? e + (n - e) * t / 60 : t < 180 ? n : t < 240 ? e + (n - e) * (240 - t) / 60 : e) * 255;
}
const Bn = (t) => () => t;
function Ea(t, e) {
  return function(n) {
    return t + n * e;
  };
}
function Ra(t, e, n) {
  return t = Math.pow(t, n), e = Math.pow(e, n) - t, n = 1 / n, function(r) {
    return Math.pow(t + r * e, n);
  };
}
function Ta(t) {
  return (t = +t) == 1 ? Zn : function(e, n) {
    return n - e ? Ra(e, n, t) : Bn(isNaN(e) ? n : e);
  };
}
function Zn(t, e) {
  var n = e - t;
  return n ? Ea(t, n) : Bn(isNaN(t) ? e : t);
}
const mn = function t(e) {
  var n = Ta(e);
  function r(i, a) {
    var o = n((i = De(i)).r, (a = De(a)).r), s = n(i.g, a.g), l = n(i.b, a.b), h = Zn(i.opacity, a.opacity);
    return function(c) {
      return i.r = o(c), i.g = s(c), i.b = l(c), i.opacity = h(c), i + "";
    };
  }
  return r.gamma = t, r;
}(1);
function wt(t, e) {
  return t = +t, e = +e, function(n) {
    return t * (1 - n) + e * n;
  };
}
var ke = /[-+]?(?:\d+\.?\d*|\.?\d+)(?:[eE][-+]?\d+)?/g, Oe = new RegExp(ke.source, "g");
function Sa(t) {
  return function() {
    return t;
  };
}
function Ca(t) {
  return function(e) {
    return t(e) + "";
  };
}
function Oa(t, e) {
  var n = ke.lastIndex = Oe.lastIndex = 0, r, i, a, o = -1, s = [], l = [];
  for (t = t + "", e = e + ""; (r = ke.exec(t)) && (i = Oe.exec(e)); )
    (a = i.index) > n && (a = e.slice(n, a), s[o] ? s[o] += a : s[++o] = a), (r = r[0]) === (i = i[0]) ? s[o] ? s[o] += i : s[++o] = i : (s[++o] = null, l.push({ i: o, x: wt(r, i) })), n = Oe.lastIndex;
  return n < e.length && (a = e.slice(n), s[o] ? s[o] += a : s[++o] = a), s.length < 2 ? l[0] ? Ca(l[0].x) : Sa(e) : (e = l.length, function(h) {
    for (var c = 0, _; c < e; ++c)
      s[(_ = l[c]).i] = _.x(h);
    return s.join("");
  });
}
var yn = 180 / Math.PI, $e = {
  translateX: 0,
  translateY: 0,
  rotate: 0,
  skewX: 0,
  scaleX: 1,
  scaleY: 1
};
function Jn(t, e, n, r, i, a) {
  var o, s, l;
  return (o = Math.sqrt(t * t + e * e)) && (t /= o, e /= o), (l = t * n + e * r) && (n -= t * l, r -= e * l), (s = Math.sqrt(n * n + r * r)) && (n /= s, r /= s, l /= s), t * r < e * n && (t = -t, e = -e, l = -l, o = -o), {
    translateX: i,
    translateY: a,
    rotate: Math.atan2(e, t) * yn,
    skewX: Math.atan(l) * yn,
    scaleX: o,
    scaleY: s
  };
}
var ne;
function Aa(t) {
  const e = new (typeof DOMMatrix == "function" ? DOMMatrix : WebKitCSSMatrix)(t + "");
  return e.isIdentity ? $e : Jn(e.a, e.b, e.c, e.d, e.e, e.f);
}
function Na(t) {
  return t == null || (ne || (ne = document.createElementNS("http://www.w3.org/2000/svg", "g")), ne.setAttribute("transform", t), !(t = ne.transform.baseVal.consolidate())) ? $e : (t = t.matrix, Jn(t.a, t.b, t.c, t.d, t.e, t.f));
}
function Qn(t, e, n, r) {
  function i(h) {
    return h.length ? h.pop() + " " : "";
  }
  function a(h, c, _, p, y, O) {
    if (h !== _ || c !== p) {
      var T = y.push("translate(", null, e, null, n);
      O.push({ i: T - 4, x: wt(h, _) }, { i: T - 2, x: wt(c, p) });
    } else
      (_ || p) && y.push("translate(" + _ + e + p + n);
  }
  function o(h, c, _, p) {
    h !== c ? (h - c > 180 ? c += 360 : c - h > 180 && (h += 360), p.push({ i: _.push(i(_) + "rotate(", null, r) - 2, x: wt(h, c) })) : c && _.push(i(_) + "rotate(" + c + r);
  }
  function s(h, c, _, p) {
    h !== c ? p.push({ i: _.push(i(_) + "skewX(", null, r) - 2, x: wt(h, c) }) : c && _.push(i(_) + "skewX(" + c + r);
  }
  function l(h, c, _, p, y, O) {
    if (h !== _ || c !== p) {
      var T = y.push(i(y) + "scale(", null, ",", null, ")");
      O.push({ i: T - 4, x: wt(h, _) }, { i: T - 2, x: wt(c, p) });
    } else
      (_ !== 1 || p !== 1) && y.push(i(y) + "scale(" + _ + "," + p + ")");
  }
  return function(h, c) {
    var _ = [], p = [];
    return h = t(h), c = t(c), a(h.translateX, h.translateY, c.translateX, c.translateY, _, p), o(h.rotate, c.rotate, _, p), s(h.skewX, c.skewX, _, p), l(h.scaleX, h.scaleY, c.scaleX, c.scaleY, _, p), h = c = null, function(y) {
      for (var O = -1, T = p.length, M; ++O < T; )
        _[(M = p[O]).i] = M.x(y);
      return _.join("");
    };
  };
}
var za = Qn(Aa, "px, ", "px)", "deg)"), Pa = Qn(Na, ", ", ")", ")"), Ma = 1e-12;
function xn(t) {
  return ((t = Math.exp(t)) + 1 / t) / 2;
}
function Ia(t) {
  return ((t = Math.exp(t)) - 1 / t) / 2;
}
function Da(t) {
  return ((t = Math.exp(2 * t)) - 1) / (t + 1);
}
const ka = function t(e, n, r) {
  function i(a, o) {
    var s = a[0], l = a[1], h = a[2], c = o[0], _ = o[1], p = o[2], y = c - s, O = _ - l, T = y * y + O * O, M, w;
    if (T < Ma)
      w = Math.log(p / h) / e, M = function(I) {
        return [
          s + I * y,
          l + I * O,
          h * Math.exp(e * I * w)
        ];
      };
    else {
      var $ = Math.sqrt(T), C = (p * p - h * h + r * T) / (2 * h * n * $), V = (p * p - h * h - r * T) / (2 * p * n * $), J = Math.log(Math.sqrt(C * C + 1) - C), j = Math.log(Math.sqrt(V * V + 1) - V);
      w = (j - J) / e, M = function(I) {
        var A = I * w, H = xn(J), rt = h / (n * $) * (H * Da(e * A + J) - Ia(J));
        return [
          s + rt * y,
          l + rt * O,
          h * H / xn(e * A + J)
        ];
      };
    }
    return M.duration = w * 1e3 * e / Math.SQRT2, M;
  }
  return i.rho = function(a) {
    var o = Math.max(1e-3, +a), s = o * o, l = s * s;
    return t(o, s, l);
  }, i;
}(Math.SQRT2, 2, 4);
var kt = 0, Xt = 0, Yt = 0, tr = 1e3, fe, Ht, he = 0, At = 0, ve = 0, Kt = typeof performance == "object" && performance.now ? performance : Date, er = typeof window == "object" && window.requestAnimationFrame ? window.requestAnimationFrame.bind(window) : function(t) {
  setTimeout(t, 17);
};
function qe() {
  return At || (er($a), At = Kt.now() + ve);
}
function $a() {
  At = 0;
}
function de() {
  this._call = this._time = this._next = null;
}
de.prototype = nr.prototype = {
  constructor: de,
  restart: function(t, e, n) {
    if (typeof t != "function")
      throw new TypeError("callback is not a function");
    n = (n == null ? qe() : +n) + (e == null ? 0 : +e), !this._next && Ht !== this && (Ht ? Ht._next = this : fe = this, Ht = this), this._call = t, this._time = n, Le();
  },
  stop: function() {
    this._call && (this._call = null, this._time = 1 / 0, Le());
  }
};
function nr(t, e, n) {
  var r = new de();
  return r.restart(t, e, n), r;
}
function La() {
  qe(), ++kt;
  for (var t = fe, e; t; )
    (e = At - t._time) >= 0 && t._call.call(void 0, e), t = t._next;
  --kt;
}
function wn() {
  At = (he = Kt.now()) + ve, kt = Xt = 0;
  try {
    La();
  } finally {
    kt = 0, Fa(), At = 0;
  }
}
function Ya() {
  var t = Kt.now(), e = t - he;
  e > tr && (ve -= e, he = t);
}
function Fa() {
  for (var t, e = fe, n, r = 1 / 0; e; )
    e._call ? (r > e._time && (r = e._time), t = e, e = e._next) : (n = e._next, e._next = null, e = t ? t._next = n : fe = n);
  Ht = t, Le(r);
}
function Le(t) {
  if (!kt) {
    Xt && (Xt = clearTimeout(Xt));
    var e = t - At;
    e > 24 ? (t < 1 / 0 && (Xt = setTimeout(wn, t - Kt.now() - ve)), Yt && (Yt = clearInterval(Yt))) : (Yt || (he = Kt.now(), Yt = setInterval(Ya, tr)), kt = 1, er(wn));
  }
}
function bn(t, e, n) {
  var r = new de();
  return e = e == null ? 0 : +e, r.restart((i) => {
    r.stop(), t(i + e);
  }, e, n), r;
}
var Xa = Ue("start", "end", "cancel", "interrupt"), Ha = [], rr = 0, En = 1, Ye = 2, ae = 3, Rn = 4, Fe = 5, oe = 6;
function me(t, e, n, r, i, a) {
  var o = t.__transition;
  if (!o)
    t.__transition = {};
  else if (n in o)
    return;
  ja(t, n, {
    name: e,
    index: r,
    // For context during callback.
    group: i,
    // For context during callback.
    on: Xa,
    tween: Ha,
    time: a.time,
    delay: a.delay,
    duration: a.duration,
    ease: a.ease,
    timer: null,
    state: rr
  });
}
function We(t, e) {
  var n = lt(t, e);
  if (n.state > rr)
    throw new Error("too late; already scheduled");
  return n;
}
function dt(t, e) {
  var n = lt(t, e);
  if (n.state > ae)
    throw new Error("too late; already running");
  return n;
}
function lt(t, e) {
  var n = t.__transition;
  if (!n || !(n = n[e]))
    throw new Error("transition not found");
  return n;
}
function ja(t, e, n) {
  var r = t.__transition, i;
  r[e] = n, n.timer = nr(a, 0, n.time);
  function a(h) {
    n.state = En, n.timer.restart(o, n.delay, n.time), n.delay <= h && o(h - n.delay);
  }
  function o(h) {
    var c, _, p, y;
    if (n.state !== En)
      return l();
    for (c in r)
      if (y = r[c], y.name === n.name) {
        if (y.state === ae)
          return bn(o);
        y.state === Rn ? (y.state = oe, y.timer.stop(), y.on.call("interrupt", t, t.__data__, y.index, y.group), delete r[c]) : +c < e && (y.state = oe, y.timer.stop(), y.on.call("cancel", t, t.__data__, y.index, y.group), delete r[c]);
      }
    if (bn(function() {
      n.state === ae && (n.state = Rn, n.timer.restart(s, n.delay, n.time), s(h));
    }), n.state = Ye, n.on.call("start", t, t.__data__, n.index, n.group), n.state === Ye) {
      for (n.state = ae, i = new Array(p = n.tween.length), c = 0, _ = -1; c < p; ++c)
        (y = n.tween[c].value.call(t, t.__data__, n.index, n.group)) && (i[++_] = y);
      i.length = _ + 1;
    }
  }
  function s(h) {
    for (var c = h < n.duration ? n.ease.call(null, h / n.duration) : (n.timer.restart(l), n.state = Fe, 1), _ = -1, p = i.length; ++_ < p; )
      i[_].call(t, c);
    n.state === Fe && (n.on.call("end", t, t.__data__, n.index, n.group), l());
  }
  function l() {
    n.state = oe, n.timer.stop(), delete r[e];
    for (var h in r)
      return;
    delete t.__transition;
  }
}
function se(t, e) {
  var n = t.__transition, r, i, a = !0, o;
  if (n) {
    e = e == null ? null : e + "";
    for (o in n) {
      if ((r = n[o]).name !== e) {
        a = !1;
        continue;
      }
      i = r.state > Ye && r.state < Fe, r.state = oe, r.timer.stop(), r.on.call(i ? "interrupt" : "cancel", t, t.__data__, r.index, r.group), delete n[o];
    }
    a && delete t.__transition;
  }
}
function Ua(t) {
  return this.each(function() {
    se(this, t);
  });
}
function Va(t, e) {
  var n, r;
  return function() {
    var i = dt(this, t), a = i.tween;
    if (a !== n) {
      r = n = a;
      for (var o = 0, s = r.length; o < s; ++o)
        if (r[o].name === e) {
          r = r.slice(), r.splice(o, 1);
          break;
        }
    }
    i.tween = r;
  };
}
function qa(t, e, n) {
  var r, i;
  if (typeof n != "function")
    throw new Error();
  return function() {
    var a = dt(this, t), o = a.tween;
    if (o !== r) {
      i = (r = o).slice();
      for (var s = { name: e, value: n }, l = 0, h = i.length; l < h; ++l)
        if (i[l].name === e) {
          i[l] = s;
          break;
        }
      l === h && i.push(s);
    }
    a.tween = i;
  };
}
function Wa(t, e) {
  var n = this._id;
  if (t += "", arguments.length < 2) {
    for (var r = lt(this.node(), n).tween, i = 0, a = r.length, o; i < a; ++i)
      if ((o = r[i]).name === t)
        return o.value;
    return null;
  }
  return this.each((e == null ? Va : qa)(n, t, e));
}
function Ge(t, e, n) {
  var r = t._id;
  return t.each(function() {
    var i = dt(this, r);
    (i.value || (i.value = {}))[e] = n.apply(this, arguments);
  }), function(i) {
    return lt(i, r).value[e];
  };
}
function ir(t, e) {
  var n;
  return (typeof e == "number" ? wt : e instanceof Gt ? mn : (n = Gt(e)) ? (e = n, mn) : Oa)(t, e);
}
function Ga(t) {
  return function() {
    this.removeAttribute(t);
  };
}
function Ka(t) {
  return function() {
    this.removeAttributeNS(t.space, t.local);
  };
}
function Ba(t, e, n) {
  var r, i = n + "", a;
  return function() {
    var o = this.getAttribute(t);
    return o === i ? null : o === r ? a : a = e(r = o, n);
  };
}
function Za(t, e, n) {
  var r, i = n + "", a;
  return function() {
    var o = this.getAttributeNS(t.space, t.local);
    return o === i ? null : o === r ? a : a = e(r = o, n);
  };
}
function Ja(t, e, n) {
  var r, i, a;
  return function() {
    var o, s = n(this), l;
    return s == null ? void this.removeAttribute(t) : (o = this.getAttribute(t), l = s + "", o === l ? null : o === r && l === i ? a : (i = l, a = e(r = o, s)));
  };
}
function Qa(t, e, n) {
  var r, i, a;
  return function() {
    var o, s = n(this), l;
    return s == null ? void this.removeAttributeNS(t.space, t.local) : (o = this.getAttributeNS(t.space, t.local), l = s + "", o === l ? null : o === r && l === i ? a : (i = l, a = e(r = o, s)));
  };
}
function to(t, e) {
  var n = ge(t), r = n === "transform" ? Pa : ir;
  return this.attrTween(t, typeof e == "function" ? (n.local ? Qa : Ja)(n, r, Ge(this, "attr." + t, e)) : e == null ? (n.local ? Ka : Ga)(n) : (n.local ? Za : Ba)(n, r, e));
}
function eo(t, e) {
  return function(n) {
    this.setAttribute(t, e.call(this, n));
  };
}
function no(t, e) {
  return function(n) {
    this.setAttributeNS(t.space, t.local, e.call(this, n));
  };
}
function ro(t, e) {
  var n, r;
  function i() {
    var a = e.apply(this, arguments);
    return a !== r && (n = (r = a) && no(t, a)), n;
  }
  return i._value = e, i;
}
function io(t, e) {
  var n, r;
  function i() {
    var a = e.apply(this, arguments);
    return a !== r && (n = (r = a) && eo(t, a)), n;
  }
  return i._value = e, i;
}
function ao(t, e) {
  var n = "attr." + t;
  if (arguments.length < 2)
    return (n = this.tween(n)) && n._value;
  if (e == null)
    return this.tween(n, null);
  if (typeof e != "function")
    throw new Error();
  var r = ge(t);
  return this.tween(n, (r.local ? ro : io)(r, e));
}
function oo(t, e) {
  return function() {
    We(this, t).delay = +e.apply(this, arguments);
  };
}
function so(t, e) {
  return e = +e, function() {
    We(this, t).delay = e;
  };
}
function uo(t) {
  var e = this._id;
  return arguments.length ? this.each((typeof t == "function" ? oo : so)(e, t)) : lt(this.node(), e).delay;
}
function co(t, e) {
  return function() {
    dt(this, t).duration = +e.apply(this, arguments);
  };
}
function lo(t, e) {
  return e = +e, function() {
    dt(this, t).duration = e;
  };
}
function fo(t) {
  var e = this._id;
  return arguments.length ? this.each((typeof t == "function" ? co : lo)(e, t)) : lt(this.node(), e).duration;
}
function ho(t, e) {
  if (typeof e != "function")
    throw new Error();
  return function() {
    dt(this, t).ease = e;
  };
}
function po(t) {
  var e = this._id;
  return arguments.length ? this.each(ho(e, t)) : lt(this.node(), e).ease;
}
function _o(t, e) {
  return function() {
    var n = e.apply(this, arguments);
    if (typeof n != "function")
      throw new Error();
    dt(this, t).ease = n;
  };
}
function go(t) {
  if (typeof t != "function")
    throw new Error();
  return this.each(_o(this._id, t));
}
function vo(t) {
  typeof t != "function" && (t = Ln(t));
  for (var e = this._groups, n = e.length, r = new Array(n), i = 0; i < n; ++i)
    for (var a = e[i], o = a.length, s = r[i] = [], l, h = 0; h < o; ++h)
      (l = a[h]) && t.call(l, l.__data__, h, a) && s.push(l);
  return new mt(r, this._parents, this._name, this._id);
}
function mo(t) {
  if (t._id !== this._id)
    throw new Error();
  for (var e = this._groups, n = t._groups, r = e.length, i = n.length, a = Math.min(r, i), o = new Array(r), s = 0; s < a; ++s)
    for (var l = e[s], h = n[s], c = l.length, _ = o[s] = new Array(c), p, y = 0; y < c; ++y)
      (p = l[y] || h[y]) && (_[y] = p);
  for (; s < r; ++s)
    o[s] = e[s];
  return new mt(o, this._parents, this._name, this._id);
}
function yo(t) {
  return (t + "").trim().split(/^|\s+/).every(function(e) {
    var n = e.indexOf(".");
    return n >= 0 && (e = e.slice(0, n)), !e || e === "start";
  });
}
function xo(t, e, n) {
  var r, i, a = yo(e) ? We : dt;
  return function() {
    var o = a(this, t), s = o.on;
    s !== r && (i = (r = s).copy()).on(e, n), o.on = i;
  };
}
function wo(t, e) {
  var n = this._id;
  return arguments.length < 2 ? lt(this.node(), n).on.on(t) : this.each(xo(n, t, e));
}
function bo(t) {
  return function() {
    var e = this.parentNode;
    for (var n in this.__transition)
      if (+n !== t)
        return;
    e && e.removeChild(this);
  };
}
function Eo() {
  return this.on("end.remove", bo(this._id));
}
function Ro(t) {
  var e = this._name, n = this._id;
  typeof t != "function" && (t = He(t));
  for (var r = this._groups, i = r.length, a = new Array(i), o = 0; o < i; ++o)
    for (var s = r[o], l = s.length, h = a[o] = new Array(l), c, _, p = 0; p < l; ++p)
      (c = s[p]) && (_ = t.call(c, c.__data__, p, s)) && ("__data__" in c && (_.__data__ = c.__data__), h[p] = _, me(h[p], e, n, p, h, lt(c, n)));
  return new mt(a, this._parents, e, n);
}
function To(t) {
  var e = this._name, n = this._id;
  typeof t != "function" && (t = $n(t));
  for (var r = this._groups, i = r.length, a = [], o = [], s = 0; s < i; ++s)
    for (var l = r[s], h = l.length, c, _ = 0; _ < h; ++_)
      if (c = l[_]) {
        for (var p = t.call(c, c.__data__, _, l), y, O = lt(c, n), T = 0, M = p.length; T < M; ++T)
          (y = p[T]) && me(y, e, n, T, p, O);
        a.push(p), o.push(c);
      }
  return new mt(a, o, e, n);
}
var So = Bt.prototype.constructor;
function Co() {
  return new So(this._groups, this._parents);
}
function Oo(t, e) {
  var n, r, i;
  return function() {
    var a = Dt(this, t), o = (this.style.removeProperty(t), Dt(this, t));
    return a === o ? null : a === n && o === r ? i : i = e(n = a, r = o);
  };
}
function ar(t) {
  return function() {
    this.style.removeProperty(t);
  };
}
function Ao(t, e, n) {
  var r, i = n + "", a;
  return function() {
    var o = Dt(this, t);
    return o === i ? null : o === r ? a : a = e(r = o, n);
  };
}
function No(t, e, n) {
  var r, i, a;
  return function() {
    var o = Dt(this, t), s = n(this), l = s + "";
    return s == null && (l = s = (this.style.removeProperty(t), Dt(this, t))), o === l ? null : o === r && l === i ? a : (i = l, a = e(r = o, s));
  };
}
function zo(t, e) {
  var n, r, i, a = "style." + e, o = "end." + a, s;
  return function() {
    var l = dt(this, t), h = l.on, c = l.value[a] == null ? s || (s = ar(e)) : void 0;
    (h !== n || i !== c) && (r = (n = h).copy()).on(o, i = c), l.on = r;
  };
}
function Po(t, e, n) {
  var r = (t += "") == "transform" ? za : ir;
  return e == null ? this.styleTween(t, Oo(t, r)).on("end.style." + t, ar(t)) : typeof e == "function" ? this.styleTween(t, No(t, r, Ge(this, "style." + t, e))).each(zo(this._id, t)) : this.styleTween(t, Ao(t, r, e), n).on("end.style." + t, null);
}
function Mo(t, e, n) {
  return function(r) {
    this.style.setProperty(t, e.call(this, r), n);
  };
}
function Io(t, e, n) {
  var r, i;
  function a() {
    var o = e.apply(this, arguments);
    return o !== i && (r = (i = o) && Mo(t, o, n)), r;
  }
  return a._value = e, a;
}
function Do(t, e, n) {
  var r = "style." + (t += "");
  if (arguments.length < 2)
    return (r = this.tween(r)) && r._value;
  if (e == null)
    return this.tween(r, null);
  if (typeof e != "function")
    throw new Error();
  return this.tween(r, Io(t, e, n ?? ""));
}
function ko(t) {
  return function() {
    this.textContent = t;
  };
}
function $o(t) {
  return function() {
    var e = t(this);
    this.textContent = e ?? "";
  };
}
function Lo(t) {
  return this.tween("text", typeof t == "function" ? $o(Ge(this, "text", t)) : ko(t == null ? "" : t + ""));
}
function Yo(t) {
  return function(e) {
    this.textContent = t.call(this, e);
  };
}
function Fo(t) {
  var e, n;
  function r() {
    var i = t.apply(this, arguments);
    return i !== n && (e = (n = i) && Yo(i)), e;
  }
  return r._value = t, r;
}
function Xo(t) {
  var e = "text";
  if (arguments.length < 1)
    return (e = this.tween(e)) && e._value;
  if (t == null)
    return this.tween(e, null);
  if (typeof t != "function")
    throw new Error();
  return this.tween(e, Fo(t));
}
function Ho() {
  for (var t = this._name, e = this._id, n = or(), r = this._groups, i = r.length, a = 0; a < i; ++a)
    for (var o = r[a], s = o.length, l, h = 0; h < s; ++h)
      if (l = o[h]) {
        var c = lt(l, e);
        me(l, t, n, h, o, {
          time: c.time + c.delay + c.duration,
          delay: 0,
          duration: c.duration,
          ease: c.ease
        });
      }
  return new mt(r, this._parents, t, n);
}
function jo() {
  var t, e, n = this, r = n._id, i = n.size();
  return new Promise(function(a, o) {
    var s = { value: o }, l = { value: function() {
      --i === 0 && a();
    } };
    n.each(function() {
      var h = dt(this, r), c = h.on;
      c !== t && (e = (t = c).copy(), e._.cancel.push(s), e._.interrupt.push(s), e._.end.push(l)), h.on = e;
    }), i === 0 && a();
  });
}
var Uo = 0;
function mt(t, e, n, r) {
  this._groups = t, this._parents = e, this._name = n, this._id = r;
}
function or() {
  return ++Uo;
}
var gt = Bt.prototype;
mt.prototype = {
  constructor: mt,
  select: Ro,
  selectAll: To,
  selectChild: gt.selectChild,
  selectChildren: gt.selectChildren,
  filter: vo,
  merge: mo,
  selection: Co,
  transition: Ho,
  call: gt.call,
  nodes: gt.nodes,
  node: gt.node,
  size: gt.size,
  empty: gt.empty,
  each: gt.each,
  on: wo,
  attr: to,
  attrTween: ao,
  style: Po,
  styleTween: Do,
  text: Lo,
  textTween: Xo,
  remove: Eo,
  tween: Wa,
  delay: uo,
  duration: fo,
  ease: po,
  easeVarying: go,
  end: jo,
  [Symbol.iterator]: gt[Symbol.iterator]
};
function Vo(t) {
  return ((t *= 2) <= 1 ? t * t * t : (t -= 2) * t * t + 2) / 2;
}
var qo = {
  time: null,
  // Set on use.
  delay: 0,
  duration: 250,
  ease: Vo
};
function Wo(t, e) {
  for (var n; !(n = t.__transition) || !(n = n[e]); )
    if (!(t = t.parentNode))
      throw new Error(`transition ${e} not found`);
  return n;
}
function Go(t) {
  var e, n;
  t instanceof mt ? (e = t._id, t = t._name) : (e = or(), (n = qo).time = qe(), t = t == null ? null : t + "");
  for (var r = this._groups, i = r.length, a = 0; a < i; ++a)
    for (var o = r[a], s = o.length, l, h = 0; h < s; ++h)
      (l = o[h]) && me(l, t, e, h, o, n || Wo(l, e));
  return new mt(r, this._parents, t, e);
}
Bt.prototype.interrupt = Ua;
Bt.prototype.transition = Go;
const re = (t) => () => t;
function Ko(t, {
  sourceEvent: e,
  target: n,
  transform: r,
  dispatch: i
}) {
  Object.defineProperties(this, {
    type: { value: t, enumerable: !0, configurable: !0 },
    sourceEvent: { value: e, enumerable: !0, configurable: !0 },
    target: { value: n, enumerable: !0, configurable: !0 },
    transform: { value: r, enumerable: !0, configurable: !0 },
    _: { value: i }
  });
}
function vt(t, e, n) {
  this.k = t, this.x = e, this.y = n;
}
vt.prototype = {
  constructor: vt,
  scale: function(t) {
    return t === 1 ? this : new vt(this.k * t, this.x, this.y);
  },
  translate: function(t, e) {
    return t === 0 & e === 0 ? this : new vt(this.k, this.x + this.k * t, this.y + this.k * e);
  },
  apply: function(t) {
    return [t[0] * this.k + this.x, t[1] * this.k + this.y];
  },
  applyX: function(t) {
    return t * this.k + this.x;
  },
  applyY: function(t) {
    return t * this.k + this.y;
  },
  invert: function(t) {
    return [(t[0] - this.x) / this.k, (t[1] - this.y) / this.k];
  },
  invertX: function(t) {
    return (t - this.x) / this.k;
  },
  invertY: function(t) {
    return (t - this.y) / this.k;
  },
  rescaleX: function(t) {
    return t.copy().domain(t.range().map(this.invertX, this).map(t.invert, t));
  },
  rescaleY: function(t) {
    return t.copy().domain(t.range().map(this.invertY, this).map(t.invert, t));
  },
  toString: function() {
    return "translate(" + this.x + "," + this.y + ") scale(" + this.k + ")";
  }
};
var Ut = new vt(1, 0, 0);
vt.prototype;
function Ae(t) {
  t.stopImmediatePropagation();
}
function Ft(t) {
  t.preventDefault(), t.stopImmediatePropagation();
}
function Bo(t) {
  return (!t.ctrlKey || t.type === "wheel") && !t.button;
}
function Zo() {
  var t = this;
  return t instanceof SVGElement ? (t = t.ownerSVGElement || t, t.hasAttribute("viewBox") ? (t = t.viewBox.baseVal, [[t.x, t.y], [t.x + t.width, t.y + t.height]]) : [[0, 0], [t.width.baseVal.value, t.height.baseVal.value]]) : [[0, 0], [t.clientWidth, t.clientHeight]];
}
function Tn() {
  return this.__zoom || Ut;
}
function Jo(t) {
  return -t.deltaY * (t.deltaMode === 1 ? 0.05 : t.deltaMode ? 1 : 2e-3) * (t.ctrlKey ? 10 : 1);
}
function Qo() {
  return navigator.maxTouchPoints || "ontouchstart" in this;
}
function ts(t, e, n) {
  var r = t.invertX(e[0][0]) - n[0][0], i = t.invertX(e[1][0]) - n[1][0], a = t.invertY(e[0][1]) - n[0][1], o = t.invertY(e[1][1]) - n[1][1];
  return t.translate(
    i > r ? (r + i) / 2 : Math.min(0, r) || Math.max(0, i),
    o > a ? (a + o) / 2 : Math.min(0, a) || Math.max(0, o)
  );
}
function Sn() {
  var t = Bo, e = Zo, n = ts, r = Jo, i = Qo, a = [0, 1 / 0], o = [[-1 / 0, -1 / 0], [1 / 0, 1 / 0]], s = 250, l = ka, h = Ue("start", "zoom", "end"), c, _, p, y = 500, O = 150, T = 0, M = 10;
  function w(f) {
    f.property("__zoom", Tn).on("wheel.zoom", A, { passive: !1 }).on("mousedown.zoom", H).on("dblclick.zoom", rt).filter(i).on("touchstart.zoom", ut).on("touchmove.zoom", P).on("touchend.zoom touchcancel.zoom", L).style("-webkit-tap-highlight-color", "rgba(0,0,0,0)");
  }
  w.transform = function(f, v, d, m) {
    var b = f.selection ? f.selection() : f;
    b.property("__zoom", Tn), f !== b ? J(f, v, d, m) : b.interrupt().each(function() {
      j(this, arguments).event(m).start().zoom(null, typeof v == "function" ? v.apply(this, arguments) : v).end();
    });
  }, w.scaleBy = function(f, v, d, m) {
    w.scaleTo(f, function() {
      var b = this.__zoom.k, E = typeof v == "function" ? v.apply(this, arguments) : v;
      return b * E;
    }, d, m);
  }, w.scaleTo = function(f, v, d, m) {
    w.transform(f, function() {
      var b = e.apply(this, arguments), E = this.__zoom, R = d == null ? V(b) : typeof d == "function" ? d.apply(this, arguments) : d, z = E.invert(R), N = typeof v == "function" ? v.apply(this, arguments) : v;
      return n(C($(E, N), R, z), b, o);
    }, d, m);
  }, w.translateBy = function(f, v, d, m) {
    w.transform(f, function() {
      return n(this.__zoom.translate(
        typeof v == "function" ? v.apply(this, arguments) : v,
        typeof d == "function" ? d.apply(this, arguments) : d
      ), e.apply(this, arguments), o);
    }, null, m);
  }, w.translateTo = function(f, v, d, m, b) {
    w.transform(f, function() {
      var E = e.apply(this, arguments), R = this.__zoom, z = m == null ? V(E) : typeof m == "function" ? m.apply(this, arguments) : m;
      return n(Ut.translate(z[0], z[1]).scale(R.k).translate(
        typeof v == "function" ? -v.apply(this, arguments) : -v,
        typeof d == "function" ? -d.apply(this, arguments) : -d
      ), E, o);
    }, m, b);
  };
  function $(f, v) {
    return v = Math.max(a[0], Math.min(a[1], v)), v === f.k ? f : new vt(v, f.x, f.y);
  }
  function C(f, v, d) {
    var m = v[0] - d[0] * f.k, b = v[1] - d[1] * f.k;
    return m === f.x && b === f.y ? f : new vt(f.k, m, b);
  }
  function V(f) {
    return [(+f[0][0] + +f[1][0]) / 2, (+f[0][1] + +f[1][1]) / 2];
  }
  function J(f, v, d, m) {
    f.on("start.zoom", function() {
      j(this, arguments).event(m).start();
    }).on("interrupt.zoom end.zoom", function() {
      j(this, arguments).event(m).end();
    }).tween("zoom", function() {
      var b = this, E = arguments, R = j(b, E).event(m), z = e.apply(b, E), N = d == null ? V(z) : typeof d == "function" ? d.apply(b, E) : d, q = Math.max(z[1][0] - z[0][0], z[1][1] - z[0][1]), X = b.__zoom, W = typeof v == "function" ? v.apply(b, E) : v, K = l(X.invert(N).concat(q / X.k), W.invert(N).concat(q / W.k));
      return function(B) {
        if (B === 1)
          B = W;
        else {
          var Q = K(B), it = q / Q[2];
          B = new vt(it, N[0] - Q[0] * it, N[1] - Q[1] * it);
        }
        R.zoom(null, B);
      };
    });
  }
  function j(f, v, d) {
    return !d && f.__zooming || new I(f, v);
  }
  function I(f, v) {
    this.that = f, this.args = v, this.active = 0, this.sourceEvent = null, this.extent = e.apply(f, v), this.taps = 0;
  }
  I.prototype = {
    event: function(f) {
      return f && (this.sourceEvent = f), this;
    },
    start: function() {
      return ++this.active === 1 && (this.that.__zooming = this, this.emit("start")), this;
    },
    zoom: function(f, v) {
      return this.mouse && f !== "mouse" && (this.mouse[1] = v.invert(this.mouse[0])), this.touch0 && f !== "touch" && (this.touch0[1] = v.invert(this.touch0[0])), this.touch1 && f !== "touch" && (this.touch1[1] = v.invert(this.touch1[0])), this.that.__zoom = v, this.emit("zoom"), this;
    },
    end: function() {
      return --this.active === 0 && (delete this.that.__zooming, this.emit("end")), this;
    },
    emit: function(f) {
      var v = nt(this.that).datum();
      h.call(
        f,
        this.that,
        new Ko(f, {
          sourceEvent: this.sourceEvent,
          target: w,
          type: f,
          transform: this.that.__zoom,
          dispatch: h
        }),
        v
      );
    }
  };
  function A(f, ...v) {
    if (!t.apply(this, arguments))
      return;
    var d = j(this, v).event(f), m = this.__zoom, b = Math.max(a[0], Math.min(a[1], m.k * Math.pow(2, r.apply(this, arguments)))), E = xt(f);
    if (d.wheel)
      (d.mouse[0][0] !== E[0] || d.mouse[0][1] !== E[1]) && (d.mouse[1] = m.invert(d.mouse[0] = E)), clearTimeout(d.wheel);
    else {
      if (m.k === b)
        return;
      d.mouse = [E, m.invert(E)], se(this), d.start();
    }
    Ft(f), d.wheel = setTimeout(R, O), d.zoom("mouse", n(C($(m, b), d.mouse[0], d.mouse[1]), d.extent, o));
    function R() {
      d.wheel = null, d.end();
    }
  }
  function H(f, ...v) {
    if (p || !t.apply(this, arguments))
      return;
    var d = f.currentTarget, m = j(this, v, !0).event(f), b = nt(f.view).on("mousemove.zoom", N, !0).on("mouseup.zoom", q, !0), E = xt(f, d), R = f.clientX, z = f.clientY;
    ca(f.view), Ae(f), m.mouse = [E, this.__zoom.invert(E)], se(this), m.start();
    function N(X) {
      if (Ft(X), !m.moved) {
        var W = X.clientX - R, K = X.clientY - z;
        m.moved = W * W + K * K > T;
      }
      m.event(X).zoom("mouse", n(C(m.that.__zoom, m.mouse[0] = xt(X, d), m.mouse[1]), m.extent, o));
    }
    function q(X) {
      b.on("mousemove.zoom mouseup.zoom", null), la(X.view, m.moved), Ft(X), m.event(X).end();
    }
  }
  function rt(f, ...v) {
    if (t.apply(this, arguments)) {
      var d = this.__zoom, m = xt(f.changedTouches ? f.changedTouches[0] : f, this), b = d.invert(m), E = d.k * (f.shiftKey ? 0.5 : 2), R = n(C($(d, E), m, b), e.apply(this, v), o);
      Ft(f), s > 0 ? nt(this).transition().duration(s).call(J, R, m, f) : nt(this).call(w.transform, R, m, f);
    }
  }
  function ut(f, ...v) {
    if (t.apply(this, arguments)) {
      var d = f.touches, m = d.length, b = j(this, v, f.changedTouches.length === m).event(f), E, R, z, N;
      for (Ae(f), R = 0; R < m; ++R)
        z = d[R], N = xt(z, this), N = [N, this.__zoom.invert(N), z.identifier], b.touch0 ? !b.touch1 && b.touch0[2] !== N[2] && (b.touch1 = N, b.taps = 0) : (b.touch0 = N, E = !0, b.taps = 1 + !!c);
      c && (c = clearTimeout(c)), E && (b.taps < 2 && (_ = N[0], c = setTimeout(function() {
        c = null;
      }, y)), se(this), b.start());
    }
  }
  function P(f, ...v) {
    if (this.__zooming) {
      var d = j(this, v).event(f), m = f.changedTouches, b = m.length, E, R, z, N;
      for (Ft(f), E = 0; E < b; ++E)
        R = m[E], z = xt(R, this), d.touch0 && d.touch0[2] === R.identifier ? d.touch0[0] = z : d.touch1 && d.touch1[2] === R.identifier && (d.touch1[0] = z);
      if (R = d.that.__zoom, d.touch1) {
        var q = d.touch0[0], X = d.touch0[1], W = d.touch1[0], K = d.touch1[1], B = (B = W[0] - q[0]) * B + (B = W[1] - q[1]) * B, Q = (Q = K[0] - X[0]) * Q + (Q = K[1] - X[1]) * Q;
        R = $(R, Math.sqrt(B / Q)), z = [(q[0] + W[0]) / 2, (q[1] + W[1]) / 2], N = [(X[0] + K[0]) / 2, (X[1] + K[1]) / 2];
      } else if (d.touch0)
        z = d.touch0[0], N = d.touch0[1];
      else
        return;
      d.zoom("touch", n(C(R, z, N), d.extent, o));
    }
  }
  function L(f, ...v) {
    if (this.__zooming) {
      var d = j(this, v).event(f), m = f.changedTouches, b = m.length, E, R;
      for (Ae(f), p && clearTimeout(p), p = setTimeout(function() {
        p = null;
      }, y), E = 0; E < b; ++E)
        R = m[E], d.touch0 && d.touch0[2] === R.identifier ? delete d.touch0 : d.touch1 && d.touch1[2] === R.identifier && delete d.touch1;
      if (d.touch1 && !d.touch0 && (d.touch0 = d.touch1, delete d.touch1), d.touch0)
        d.touch0[1] = this.__zoom.invert(d.touch0[0]);
      else if (d.end(), d.taps === 2 && (R = xt(R, this), Math.hypot(_[0] - R[0], _[1] - R[1]) < M)) {
        var z = nt(this).on("dblclick.zoom");
        z && z.apply(this, arguments);
      }
    }
  }
  return w.wheelDelta = function(f) {
    return arguments.length ? (r = typeof f == "function" ? f : re(+f), w) : r;
  }, w.filter = function(f) {
    return arguments.length ? (t = typeof f == "function" ? f : re(!!f), w) : t;
  }, w.touchable = function(f) {
    return arguments.length ? (i = typeof f == "function" ? f : re(!!f), w) : i;
  }, w.extent = function(f) {
    return arguments.length ? (e = typeof f == "function" ? f : re([[+f[0][0], +f[0][1]], [+f[1][0], +f[1][1]]]), w) : e;
  }, w.scaleExtent = function(f) {
    return arguments.length ? (a[0] = +f[0], a[1] = +f[1], w) : [a[0], a[1]];
  }, w.translateExtent = function(f) {
    return arguments.length ? (o[0][0] = +f[0][0], o[1][0] = +f[1][0], o[0][1] = +f[0][1], o[1][1] = +f[1][1], w) : [[o[0][0], o[0][1]], [o[1][0], o[1][1]]];
  }, w.constrain = function(f) {
    return arguments.length ? (n = f, w) : n;
  }, w.duration = function(f) {
    return arguments.length ? (s = +f, w) : s;
  }, w.interpolate = function(f) {
    return arguments.length ? (l = f, w) : l;
  }, w.on = function() {
    var f = h.on.apply(h, arguments);
    return f === h ? w : f;
  }, w.clickDistance = function(f) {
    return arguments.length ? (T = (f = +f) * f, w) : Math.sqrt(T);
  }, w.tapDistance = function(f) {
    return arguments.length ? (M = +f, w) : M;
  }, w;
}
const tt = {
  DEFAULT_MIN_ZOOM: 0.1,
  DEFAULT_MAX_ZOOM: 4,
  FIT_TO_VIEW_MAX_ZOOM: 1,
  ZOOM_RATIO: 0.5,
  TIME_FRAME: 500,
  INITIAL_POSITION_X: 0,
  INITIAL_POSITION_Y: 0,
  SCROLL_POS_RATIO: 3,
  LAYOUT_FALLBACK: 500,
  DEFAULT_LAYOUT: 1e6,
  INITIAL_SCROLLBAR_SIZE: 50,
  MIN_SCROLLBAR_SIZE: 15,
  SCROLL_DELTA_DIFF: 1.5,
  NODE_Y_OFFSET: 110
}, pe = {
  TOP_LEFT: "top-left",
  TOP_RIGHT: "top-right",
  BOTTOM_LEFT: "bottom-left",
  BOTTOM_CENTER: "bottom-center"
}, st = {
  ...pe,
  TOP_CENTER: "top-center",
  BOTTOM_RIGHT: "bottom-right",
  CENTER_LEFT: "center-left",
  CENTER_RIGHT: "center-right",
  CENTER_CENTER: "center-center"
}, Vt = ({
  value: t,
  min: e = 0,
  max: n
}) => Math.min(Math.max(t, e), n), es = ({
  position: t,
  svgBounds: e,
  nodeBounds: n,
  currentTranslateX: r,
  currentTranslateY: i,
  currentScale: a,
  updatedScale: o,
  customOffset: s
}) => {
  let [l, h, c] = [0, 0, 1];
  switch (o !== a && (c = o / a, r *= c, i *= c), t) {
    case st.TOP_LEFT:
      return {
        updatedX: r + ((e.x - n.x) * c + s.x),
        updatedY: i + ((e.y - n.y) * c + s.y)
      };
    case st.TOP_CENTER:
      return l = e.width / 2 - (n.x * c + n.width * c / 2) + e.x, {
        updatedX: r + l + s.x,
        updatedY: i + ((e.y - n.y) * c + s.y)
      };
    case st.TOP_RIGHT:
      return {
        updatedX: r + (e.width - (Math.abs(e.x - n.x * c) + n.width * c)) + s.x,
        updatedY: i + ((e.y - n.y) * c + s.y)
      };
    case st.CENTER_LEFT:
      return h = e.height / 2 - (Math.abs(e.y - n.y) * c + n.height * c / 2), {
        updatedX: r + ((e.x - n.x) * c + s.x),
        updatedY: i + h + s.y
      };
    case st.CENTER_CENTER:
      return l = e.width / 2 - (Math.abs(e.x - n.x) * c + n.width * c / 2), h = e.height / 2 - (Math.abs(e.y - n.y) * c + n.height * c / 2), {
        updatedX: r + l + s.x,
        updatedY: i + h + s.y
      };
    case st.CENTER_RIGHT:
      return h = e.height / 2 - (Math.abs(e.y - n.y) * c + n.height * c / 2), {
        updatedX: r + (e.width - (Math.abs(e.x - n.x) * c + n.width * c)) + s.x,
        updatedY: i + h + s.y
      };
    case st.BOTTOM_LEFT:
      return {
        updatedX: r + ((e.x - n.x) * c + s.x),
        updatedY: i + (e.height - (Math.abs(e.y - n.y) * c + n.height * c)) + s.y
      };
    case st.BOTTOM_CENTER:
      return l = e.width / 2 - (Math.abs(e.x - n.x) * c + n.width * c / 2), {
        updatedX: r + l + s.x,
        updatedY: i + (e.height - (Math.abs(e.y - n.y) * c + n.height * c)) + s.y
      };
    case st.BOTTOM_RIGHT:
      return {
        updatedX: r + (e.width - (Math.abs(e.x - n.x) * c + n.width * c)) + s.x,
        updatedY: i + (e.height - (Math.abs(e.y - n.y) * c + n.height)) + s.y
      };
    default:
      return {
        updatedX: r,
        updatedY: i
      };
  }
}, Cn = (t) => {
  const [e, n] = [Math.max(t - 0.2, 0.2), t];
  return [
    tt.INITIAL_SCROLLBAR_SIZE / e,
    tt.INITIAL_SCROLLBAR_SIZE / n
  ];
}, ns = ({
  isVertical: t,
  state: e,
  scrollDelta: n
}) => {
  const r = t ? e.verticalPos : e.horizontalPos, i = t ? e.verticalSize : e.horizontalSize, a = t ? e.verticalSizeDecrease : e.horizontalSizeDecrease, o = t ? n.deltaY : n.deltaX;
  return { scrollPos: r, scrollSize: i, scrollSizeDecrease: a, deltaValue: o };
}, On = ({
  isVertical: t = !0,
  state: e,
  scrollDelta: n,
  scrollLength: r
}) => {
  const { scrollPos: i, scrollSize: a, scrollSizeDecrease: o, deltaValue: s } = ns({
    isVertical: t,
    state: e,
    scrollDelta: n
  }), l = s / 10 || 0, h = s / 100 || 0;
  let [c, _] = [
    i + l,
    o
  ];
  if (c < 1)
    c = 0, _ -= Math.abs(h);
  else if (c + a > r) {
    const p = r - Math.max(
      tt.MIN_SCROLLBAR_SIZE,
      a - Math.abs(_)
    );
    c = Math.min(
      r - (a + _),
      p
    ), _ -= h;
  } else if (_ !== 0) {
    const p = c > r / 2;
    c = p ? r - a : 0, p ? (_ -= h, _ <= 0 && (_ = 0)) : (_ += Math.abs(h), _ >= 0 && (_ = 0));
  }
  return [c, _];
}, rs = "_dotSvgContainer_64xbd_1", is = {
  dotSvgContainer: rs
}, as = ({
  id: t = "",
  size: e = 1,
  minSize: n = 0.3,
  maxZoom: r = 4,
  gap: i = 20,
  zoomTransform: a = {
    scale: 1,
    translateX: 0,
    translateY: 0
  },
  className: o = "",
  minOpacity: s = 0.8,
  maxOpacity: l = 1,
  elementColor: h = "#afb7c7"
}) => {
  const { scale: c, translateX: _, translateY: p } = a, y = Vt({
    value: c * 10 / (r * 10),
    min: s,
    max: l
  }), O = i * c, T = e * c, M = Math.max(n, T), w = `patternId-${t}`;
  return /* @__PURE__ */ U.jsxs("svg", { className: `${o} ${is.dotSvgContainer}`, children: [
    /* @__PURE__ */ U.jsx(
      "pattern",
      {
        id: w,
        x: _ % O,
        y: p % O,
        width: O,
        height: O,
        patternUnits: "userSpaceOnUse",
        patternTransform: `translate(-${T},-${T})`,
        children: /* @__PURE__ */ U.jsx(
          "circle",
          {
            cx: M,
            cy: M,
            r: M,
            fill: h,
            opacity: y
          }
        )
      }
    ),
    /* @__PURE__ */ U.jsx("rect", { x: "0", y: "0", width: "100%", height: "100%", fill: `url(#${w})` })
  ] });
}, os = "_container_1xxyl_1", ss = "_canvasWrapper_1xxyl_8", us = "_panning_1xxyl_21", cs = "_canvas_1xxyl_8", ls = "_contentWrapper_1xxyl_39", St = {
  container: os,
  canvasWrapper: ss,
  panning: us,
  canvas: cs,
  contentWrapper: ls
}, fs = "_verticalScrollBar_17thb_1", hs = "_horizontalScrollBar_17thb_9", An = {
  verticalScrollBar: fs,
  horizontalScrollBar: hs
}, ds = Nn(
  ({
    scale: t,
    startingPosition: e,
    offset: n = { x: 0, y: 0 },
    color: r = "gray",
    thickness: i = "8px",
    minSize: a = "15px",
    verticalOffsetHeight: o,
    horizontalOffsetWidth: s,
    onScrollDeltaHandler: l,
    getContainerOffset: h
  }, c) => {
    const [_, p] = Cn(t), y = ft(null), O = ft(null), T = ft({
      isDragging: !1,
      vertical: !0,
      initialOffset: 0
    }), [M, w] = zn({
      scale: t,
      horizontalSize: p,
      horizontalPos: e ? e.x : (s ?? 0) / tt.SCROLL_POS_RATIO,
      horizontalSizeDecrease: 0,
      verticalSize: _,
      verticalPos: e ? e.y : (o ?? 0) / tt.SCROLL_POS_RATIO,
      verticalSizeDecrease: 0
    });
    Pn(c, () => ({
      resetScrollPos: () => {
        w((I) => ({
          ...I,
          horizontalPos: (s ?? 0) / tt.SCROLL_POS_RATIO,
          verticalPos: (o ?? 0) / tt.SCROLL_POS_RATIO,
          verticalSizeDecrease: 0,
          horizontalSizeDecrease: 0
        }));
      },
      onScrollDeltaChangeHandler: $,
      onMouseUp: V
    })), Mn(
      function() {
        w((A) => {
          let H;
          const [rt, ut] = Cn(t), P = t / A.scale || 0;
          return t < A.scale ? H = {
            horizontalPos: A.horizontalPos - P,
            verticalPos: A.verticalPos - P
          } : H = {
            horizontalPos: A.horizontalPos + P,
            verticalPos: A.verticalPos + P
          }, {
            ...A,
            scale: t,
            horizontalSize: ut,
            verticalSize: rt,
            ...H
          };
        });
      },
      [t]
    );
    function $(I) {
      T.current.isDragging || w((A) => {
        const [H, rt] = On({
          state: A,
          scrollDelta: I,
          scrollLength: o
        }), [ut, P] = On({
          isVertical: !1,
          state: A,
          scrollDelta: I,
          scrollLength: s
        });
        return {
          ...A,
          horizontalPos: ut,
          horizontalSizeDecrease: P,
          verticalPos: H,
          verticalSizeDecrease: rt
        };
      });
    }
    const C = jt(
      function(A, H) {
        if (!T.current.isDragging)
          return;
        A.stopPropagation();
        const { movementX: rt, movementY: ut, clientX: P, clientY: L, offsetX: f, offsetY: v } = A;
        T.current.initialOffset < 1 && (T.current.initialOffset = H ? v : f);
        const d = H ? ut : rt, m = (H ? L : P) - T.current.initialOffset - h(H), b = (N) => H ? [
          "verticalPos",
          N.verticalSize,
          o,
          n.y
        ] : [
          "horizontalPos",
          N.horizontalSize,
          s,
          n.x
        ];
        w((N) => {
          const [q, X, W, K] = b(N);
          return {
            ...N,
            [q]: Vt({
              value: m,
              min: K,
              max: W - X
            })
          };
        });
        const E = () => {
          const N = d > 0 ? d + tt.SCROLL_DELTA_DIFF : d - tt.SCROLL_DELTA_DIFF;
          return H ? { deltaX: 0, deltaY: N } : {
            deltaX: N,
            deltaY: 0
          };
        }, R = (H ? o : s) - (H ? M.verticalSize : M.horizontalSize);
        !(m <= 0 || m >= R) && d !== 0 && l(E());
      },
      // eslint-disable-next-line react-hooks/exhaustive-deps
      [o, s, n.y, n.x]
    ), V = jt(
      function() {
        T.current.isDragging = !1, T.current.initialOffset = 0, window.removeEventListener(
          "mousemove",
          (A) => C(A, T.current.vertical)
        ), window.removeEventListener("mouseup", V);
      },
      [C]
    ), J = jt(
      function() {
        window.addEventListener(
          "mousemove",
          (A) => C(A, T.current.vertical)
        ), window.addEventListener("mouseup", V);
      },
      [C, V]
    ), j = (I) => {
      I.stopPropagation();
    };
    return /* @__PURE__ */ U.jsxs(Er, { children: [
      /* @__PURE__ */ U.jsx(
        "div",
        {
          className: An.verticalScrollBar,
          style: { width: i },
          ref: y,
          children: /* @__PURE__ */ U.jsx(
            "div",
            {
              "data-id": "vertical-scrollbar",
              style: {
                insetBlockStart: `${Vt({
                  value: M.verticalPos,
                  min: n.y,
                  max: o
                })}px`,
                height: `${M.verticalSize + M.verticalSizeDecrease}px`,
                background: r,
                minHeight: a
              },
              onMouseDownCapture: (I) => {
                I.stopPropagation(), T.current = {
                  ...T.current,
                  isDragging: !0,
                  vertical: !0
                }, J();
              },
              onDragEnterCapture: j,
              onDragCapture: j,
              onDragEndCapture: j
            }
          )
        }
      ),
      /* @__PURE__ */ U.jsx(
        "div",
        {
          className: An.horizontalScrollBar,
          ref: O,
          style: { height: i },
          children: /* @__PURE__ */ U.jsx(
            "div",
            {
              "data-id": "horizontal-scrollbar",
              style: {
                insetInlineStart: `${Vt({
                  value: M.horizontalPos,
                  min: n.x,
                  max: s
                })}px`,
                width: `${M.horizontalSize + M.horizontalSizeDecrease}px`,
                background: r,
                minWidth: a
              },
              onMouseDownCapture: (I) => {
                I.stopPropagation(), T.current = {
                  ...T.current,
                  isDragging: !0,
                  vertical: !1
                }, J();
              },
              onDragEnterCapture: j,
              onDragCapture: j,
              onDragEndCapture: j
            }
          )
        }
      )
    ] });
  }
), Xe = /^((?!chrome|android).)*safari/i.test(navigator.userAgent), Ne = Xe ? 600 : 300, vs = Nn(({ children: t, ...e }, n) => {
  const r = _e.useRef(null);
  return /* @__PURE__ */ U.jsx(ps, { innerRef: n, ...e, children: /* @__PURE__ */ U.jsx(
    "div",
    {
      ref: r,
      style: { width: "max-content", height: "max-content" },
      children: t
    }
  ) });
}), ps = Rr(
  ({
    children: t,
    innerRef: e,
    minZoom: n = tt.DEFAULT_MIN_ZOOM,
    maxZoom: r = tt.DEFAULT_MAX_ZOOM,
    panOnScroll: i = !0,
    customComponents: a = [],
    renderScrollBar: o = !0,
    scrollBarConfig: s = {},
    backgroundConfig: l = {},
    onCanvasMount: h = () => {
    }
  }) => {
    const c = ft(null), _ = ft(null), p = ft(null), y = ft(null), O = ft(null), T = t.ref, M = ft(null), w = In(() => Sn().scaleExtent([n, r]), [r, n]), $ = ft(nt(p.current).call(w)), [C, V] = zn({
      translateX: 0,
      translateY: 0,
      scale: 1
    });
    Pn(e, () => ({
      scrollNodeToCenter: ({
        nodeElement: P,
        offset: L,
        scale: f,
        shouldUpdateMaxScale: v,
        maxScale: d,
        transitionDuration: m
      }) => I({
        nodeElement: P,
        offset: L,
        scale: f,
        shouldUpdateMaxScale: v,
        maxScale: d,
        transitionDuration: m,
        position: st.CENTER_CENTER
      }),
      scrollNodeHandler: I,
      scrollContentHorizontallyCenter: A,
      fitContentToView: j,
      getCanvasState: H
    })), Mn(
      function() {
        $.current = nt(p.current).call(w);
        const L = nt(y.current);
        _.current = c.current ? c.current.getBoundingClientRect() : {}, w.filter((f) => (f.type === "mousedown" && !M.current && (M.current = !0, rt()), f.ctrlKey || f.type !== "wheel")).on(
          "zoom",
          (f) => {
            var R, z;
            ((R = f.sourceEvent) == null ? void 0 : R.ctrlKey) === !1 && f.type === "zoom" && ((z = c.current) == null || z.classList.add(St.panning));
            const v = f.transform, { x: d, y: m, k: b } = v, E = y.current;
            V({ translateX: d, translateY: m, scale: b }), Xe && E ? E.style.transform = `translate(${d}px, ${m}px) scale(${b})` : L.attr("transform", v);
          }
        ), w.on("end", () => {
          var f;
          M.current = !1, (f = c.current) == null || f.classList.remove(St.panning);
        }), h({
          scrollNodeToCenter: ({
            nodeElement: f,
            offset: v,
            scale: d,
            shouldUpdateMaxScale: m,
            maxScale: b,
            transitionDuration: E
          }) => I({
            nodeElement: f,
            offset: v,
            scale: d,
            shouldUpdateMaxScale: m,
            maxScale: b,
            transitionDuration: E,
            position: st.CENTER_CENTER
          }),
          scrollNodeHandler: I,
          scrollContentHorizontallyCenter: A,
          fitContentToView: j,
          getCanvasState: H
        });
      },
      // eslint-disable-next-line react-hooks/exhaustive-deps
      []
    ), $.current.call(Sn).on(
      "wheel.zoom",
      (P) => {
        var f;
        P.preventDefault();
        const L = $.current.property("__zoom").k || 1;
        if (i && !P.ctrlKey) {
          const v = {
            deltaX: P.deltaX,
            deltaY: P.deltaY
          };
          (f = O.current) == null || f.onScrollDeltaChangeHandler(v), J(v);
        } else {
          const v = L * Math.pow(2, -P.deltaY * 0.01);
          w.scaleTo($.current, v, xt(P));
        }
      },
      { passive: !1, capture: !0 }
    );
    const J = (P) => {
      const L = $.current.property("__zoom").k || 1;
      w.translateBy(
        $.current,
        -(P.deltaX / L),
        -(P.deltaY / L)
      );
    }, j = jt(
      function({
        duration: L = 500,
        offset: f = { x: 0, y: 0 },
        scale: v,
        maxZoomLimit: d = tt.FIT_TO_VIEW_MAX_ZOOM,
        disableVerticalCenter: m = !1
      }) {
        requestIdleCallback(
          () => {
            var Tt, Qt;
            if (!T.current)
              return;
            const b = nt(p.current), E = T.current.getBoundingClientRect(), R = $.current.property("__zoom").k || 1, z = (Tt = p.current) == null ? void 0 : Tt.getBoundingClientRect(), { width: N = 0, height: q = 0 } = z || {}, X = 1 / R, W = E.width * X, K = E.height * X, B = q / K, Q = N / W, it = v ?? Vt({
              value: Math.min(
                d,
                Math.min(B, Q)
              ),
              min: n,
              max: r
            }), yt = N - W * it, Nt = q - K * it, bt = !m && B > Q, Et = yt / 2, ye = bt ? Nt / 2 : 0, Rt = Et + f.x, $t = ye + f.y, Jt = Ut.translate(Rt, $t).scale(it);
            V({ translateX: Rt, translateY: $t, scale: it }), (Qt = O.current) == null || Qt.resetScrollPos(), b.transition().duration(L).call(w.transform, Jt);
          },
          { timeout: Ne }
        );
      },
      // eslint-disable-next-line react-hooks/exhaustive-deps
      [r, n]
    ), I = ({
      nodeElement: P,
      offset: L = { x: 0, y: 0 },
      scale: f,
      shouldUpdateMaxScale: v = !0,
      maxScale: d,
      transitionDuration: m = 300,
      position: b = st.TOP_CENTER
    }) => {
      requestIdleCallback(
        () => {
          if (!P)
            return;
          const E = $.current.property("__zoom"), {
            k: R,
            x: z,
            y: N
          } = E, q = nt(p.current), W = (() => {
            const Nt = (Et) => d ? Math.min(d, Et) : Et;
            if (!f)
              return Nt(R);
            let bt = f;
            return v && (bt = Math.max(f, R)), Nt(bt);
          })(), K = p.current.getBoundingClientRect(), B = P.getBoundingClientRect(), { updatedX: Q, updatedY: it } = es({
            position: b,
            svgBounds: K,
            nodeBounds: B,
            currentTranslateX: z,
            currentTranslateY: N,
            currentScale: R,
            updatedScale: W,
            customOffset: { x: L.x, y: L.y }
          }), yt = Ut.translate(Q, it).scale(W);
          q.transition().duration(m).call(w.transform, yt);
        },
        { timeout: Ne }
      );
    }, A = ({
      offset: P = 0,
      transitionDuration: L = 300
    }) => {
      T.current && requestIdleCallback(
        () => {
          const f = $.current.property("__zoom"), { k: v, y: d } = f, m = nt(p.current), b = p.current.getBoundingClientRect(), E = T.current.getBoundingClientRect(), R = 1 / v, z = E.width * R, N = (b.width - z * v) / 2 + P;
          V({
            ...C,
            translateX: N
          });
          const q = Ut.translate(N, d).scale(v);
          m.transition().duration(L).call(w.transform, q);
        },
        { timeout: Ne }
      );
    }, H = () => ({
      canvasNode: nt(p.current),
      zoomNode: nt(y.current),
      currentPosition: $.current.property("__zoom"),
      d3Zoom: w
    }), rt = () => {
      const P = document.body;
      if (P) {
        const L = new MouseEvent("mousedown", {
          bubbles: !0,
          cancelable: !0,
          view: window
        });
        P.dispatchEvent(L);
      }
    }, ut = jt(
      function(L = !0) {
        const { x: f, y: v } = _.current;
        return L ? v : f;
      },
      []
    );
    return /* @__PURE__ */ U.jsxs("div", { className: St.container, children: [
      /* @__PURE__ */ U.jsx("div", { ref: c, className: St.canvasWrapper, children: Xe ? /* @__PURE__ */ U.jsx("div", { ref: p, className: St.canvas, children: /* @__PURE__ */ U.jsx("div", { ref: y, children: /* @__PURE__ */ U.jsx("div", { className: St.contentWrapper, children: t }) }) }) : /* @__PURE__ */ U.jsx("svg", { ref: p, className: St.canvas, children: /* @__PURE__ */ U.jsx("g", { ref: y, children: /* @__PURE__ */ U.jsx(
        "foreignObject",
        {
          x: tt.INITIAL_POSITION_X,
          y: tt.INITIAL_POSITION_Y,
          width: tt.DEFAULT_LAYOUT,
          height: tt.DEFAULT_LAYOUT,
          children: t
        }
      ) }) }) }),
      /* @__PURE__ */ U.jsx(
        as,
        {
          maxZoom: r,
          zoomTransform: C,
          ...l
        }
      ),
      o && c.current && /* @__PURE__ */ U.jsx(
        ds,
        {
          ref: O,
          scale: C.scale,
          ...s,
          verticalOffsetHeight: c.current.offsetHeight,
          horizontalOffsetWidth: c.current.offsetWidth,
          getContainerOffset: ut,
          onScrollDeltaHandler: J
        }
      ),
      a.map((P, L) => {
        const {
          component: f,
          position: v = pe.BOTTOM_LEFT,
          offset: d = { x: 0, y: 0 },
          overlap: m = !0,
          className: b = ""
        } = P;
        return /* @__PURE__ */ U.jsx(
          _s,
          {
            component: f,
            position: v,
            offset: d,
            overlap: m,
            zoomState: { ...C, minZoom: n, maxZoom: r },
            className: b
          },
          L
        );
      })
    ] });
  }
), _s = ({
  component: t,
  position: e,
  offset: n,
  overlap: r,
  zoomState: i,
  className: a
}) => {
  const o = In(() => {
    const l = Object.values(pe).includes(e) ? e : pe.BOTTOM_LEFT, [h, c] = l.split("-");
    return {
      [c]: n.x,
      [h]: n.y
    };
  }, [e, n]), s = _e.cloneElement(t, {
    zoomState: i
  });
  return /* @__PURE__ */ U.jsx(
    "div",
    {
      style: {
        position: "absolute",
        ...o,
        zIndex: r ? 20 : 1
      },
      className: a,
      children: s
    }
  );
};
export {
  as as Background,
  pe as COMPONENT_POSITIONS,
  vs as ReactInfiniteCanvas,
  st as SCROLL_NODE_POSITIONS
};
